http://bugs.freedesktop.org/show_bug.cgi?id=23162

--- dbus/Makefile.am
+++ dbus/Makefile.am
@@ -277,7 +277,7 @@
 dbus_test_SOURCES=				\
 	dbus-test-main.c
 
-dbus_test_LDADD=libdbus-internal.la $(DBUS_TEST_LIBS)
+dbus_test_LDADD=$(DBUS_TEST_LIBS) libdbus-internal.la
 dbus_test_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 ## mop up the gcov files
--- test/Makefile.am
+++ test/Makefile.am
@@ -63,17 +63,17 @@
 
 # When any programs are not linked to libdbus-internal, fix this.
 AM_CPPFLAGS=-DDBUS_STATIC_BUILD
-TEST_LIBS=$(top_builddir)/dbus/libdbus-internal.la $(DBUS_TEST_LIBS)
+TEST_LIBS=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la
 
-test_service_LDADD=libdbus-testutils.la $(TEST_LIBS)
+test_service_LDADD=$(TEST_LIBS) libdbus-testutils.la
 test_service_LDFLAGS=@R_DYNAMIC_LDFLAG@
-test_names_LDADD=libdbus-testutils.la $(TEST_LIBS)
+test_names_LDADD=$(TEST_LIBS) libdbus-testutils.la
 test_names_LDFLAGS=@R_DYNAMIC_LDFLAG@
 ## break_loader_LDADD= $(TEST_LIBS)
 ## break_loader_LDFLAGS=@R_DYNAMIC_LDFLAG@
-test_shell_service_LDADD=libdbus-testutils.la $(TEST_LIBS)
+test_shell_service_LDADD=$(TEST_LIBS) libdbus-testutils.la
 test_shell_service_LDFLAGS=@R_DYNAMIC_LDFLAG@
-shell_test_LDADD=libdbus-testutils.la $(TEST_LIBS)
+shell_test_LDADD=$(TEST_LIBS) libdbus-testutils.la
 shell_test_LDFLAGS=@R_DYNAMIC_LDFLAG@
 spawn_test_LDADD=$(TEST_LIBS)
 spawn_test_LDFLAGS=@R_DYNAMIC_LDFLAG@
--- test/name-test/Makefile.am
+++ test/name-test/Makefile.am
@@ -22,46 +22,46 @@
 test_pending_call_dispatch_SOURCES =		\
 	test-pending-call-dispatch.c
 
-test_pending_call_dispatch_LDADD=$(top_builddir)/dbus/libdbus-internal.la $(DBUS_TEST_LIBS)
+test_pending_call_dispatch_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la
 test_pending_call_dispatch_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_pending_call_timeout_SOURCES =		\
 	test-pending-call-timeout.c
 
-test_pending_call_timeout_LDADD=$(top_builddir)/dbus/libdbus-internal.la $(DBUS_TEST_LIBS)
+test_pending_call_timeout_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la
 test_pending_call_timeout_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_threads_init_SOURCES =            \
 	test-threads-init.c
 
-test_threads_init_LDADD=$(top_builddir)/dbus/libdbus-internal.la $(DBUS_TEST_LIBS)
+test_threads_init_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la
 test_threads_init_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_ids_SOURCES =            \
 	test-ids.c
 
-test_ids_LDADD=$(top_builddir)/dbus/libdbus-internal.la $(DBUS_TEST_LIBS)
+test_ids_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la
 test_ids_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_shutdown_SOURCES =            \
 	test-shutdown.c
 
 test_shutdown_CFLAGS=
-test_shutdown_LDADD=$(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_shutdown_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la
 test_shutdown_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_privserver_SOURCES =            \
 	test-privserver.c
 
 test_privserver_CFLAGS=
-test_privserver_LDADD=$(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_privserver_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la
 test_privserver_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 test_privserver_client_SOURCES =            \
 	test-privserver-client.c
 
 test_privserver_client_CFLAGS=
-test_privserver_client_LDADD=$(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la $(DBUS_TEST_LIBS)
+test_privserver_client_LDADD=$(DBUS_TEST_LIBS) $(top_builddir)/dbus/libdbus-internal.la ../libdbus-testutils.la
 test_privserver_client_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
 endif
