diff --git a/src/mesa/state_tracker/st_manager.c b/src/mesa/state_tracker/st_manager.c
index c95514c..4582bc3 100644
--- a/src/mesa/state_tracker/st_manager.c
+++ b/src/mesa/state_tracker/st_manager.c
@@ -503,8 +503,6 @@ st_context_flush(struct st_context_iface *stctxi, unsigned flags,
 {
    struct st_context *st = (struct st_context *) stctxi;
    st_flush(st, fence);
-   if (flags & ST_FLUSH_FRONT)
-      st_manager_flush_frontbuffer(st);
 }
 
 static boolean
