diff --git a/src/mesa/drivers/dri/intel/intel_tex_image.c b/src/mesa/drivers/dri/intel/intel_tex_image.c
index 68f4ff4..2d8ef91 100644
--- a/src/mesa/drivers/dri/intel/intel_tex_image.c
+++ b/src/mesa/drivers/dri/intel/intel_tex_image.c
@@ -88,6 +88,11 @@ intel_miptree_create_for_teximage(struct intel_context *intel,
 	 lastLevel = firstLevel;
       } else {
 	 lastLevel = firstLevel + _mesa_logbase2(MAX2(MAX2(width, height), depth));
+	 /* We tried to guess the last level based on the texture size, make
+	  * sure we don't go past MAX_TEXTURE_LEVELS since it's hardcoded
+	  * in many places.
+	  */
+	 lastLevel = MIN2(lastLevel, MAX_TEXTURE_LEVELS - 1);
       }
    }
 
