diff -Naur Mesa-8.1.0/src/mesa/drivers/dri/i965/gen6_clip_state.c Mesa-8.1.0_b/src/mesa/drivers/dri/i965/gen6_clip_state.c
--- Mesa-8.1.0/src/mesa/drivers/dri/i965/gen6_clip_state.c	2012-10-25 17:50:33.900301412 -0700
+++ Mesa-8.1.0_b/src/mesa/drivers/dri/i965/gen6_clip_state.c	2012-10-26 18:09:15.959711652 -0700
@@ -74,7 +74,7 @@
 	     GEN6_CLIP_MODE_NORMAL |
              nonperspective_barycentric_enable_flag |
 	     GEN6_CLIP_XY_TEST |
-	     GEN6_CLIP_GB_TEST |
+	     //GEN6_CLIP_GB_TEST |
 	     userclip << GEN6_USER_CLIP_CLIP_DISTANCES_SHIFT |
 	     depth_clamp |
 	     provoking);
diff -Naur Mesa-8.1.0/src/mesa/drivers/dri/i965/gen7_clip_state.c Mesa-8.1.0_b/src/mesa/drivers/dri/i965/gen7_clip_state.c
--- Mesa-8.1.0/src/mesa/drivers/dri/i965/gen7_clip_state.c	2012-10-25 17:50:33.900301412 -0700
+++ Mesa-8.1.0_b/src/mesa/drivers/dri/i965/gen7_clip_state.c	2012-10-26 18:10:35.230562103 -0700
@@ -101,7 +101,7 @@
 	     GEN6_CLIP_MODE_NORMAL |
              nonperspective_barycentric_enable_flag |
 	     GEN6_CLIP_XY_TEST |
-	     GEN6_CLIP_GB_TEST |
+	     //GEN6_CLIP_GB_TEST |
 	     userclip << GEN6_USER_CLIP_CLIP_DISTANCES_SHIFT |
 	     depth_clamp |
 	     provoking);
