diff --git a/src/glsl/Makefile.am b/src/glsl/Makefile.am
index 1ecc003..6cdc264 100644
--- a/src/glsl/Makefile.am
+++ b/src/glsl/Makefile.am
@@ -92,9 +92,6 @@ glsl_parser.cc glsl_parser.h: glsl_parser.yy
 BUILT_SOURCES = glsl_parser.h builtin_function.cpp
 CLEANFILES = glsl_lexer.cc glsl_parser.cc $(BUILT_SOURCES)
 
-builtin_function.cpp: $(srcdir)/builtins/profiles/* $(srcdir)/builtins/ir/* $(srcdir)/builtins/glsl/* $(srcdir)/builtins/tools/generate_builtins.py $(srcdir)/builtins/tools/texture_builtins.py builtin_compiler$(EXEEXT)
-	$(AM_V_GEN) $(PYTHON2) $(PYTHON_FLAGS) $(srcdir)/builtins/tools/generate_builtins.py ./builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
-
 glcpp/libglcpp.la:
 	cd glcpp ; $(MAKE) $(AM_MAKEFLAGS)
 
