diff --git a/src/gallium/auxiliary/util/u_format_s3tc.c b/src/gallium/auxiliary/util/u_format_s3tc.c
index 4a9dc22..31a6c19 100644
--- a/src/gallium/auxiliary/util/u_format_s3tc.c
+++ b/src/gallium/auxiliary/util/u_format_s3tc.c
@@ -120,8 +120,8 @@ util_format_s3tc_init(void)
 
    library = util_dl_open(DXTN_LIBNAME);
    if (!library) {
-      if ((force_s3tc_enable = getenv("force_s3tc_enable")) &&
-          !strcmp(force_s3tc_enable, "true")) {
+      if (1 /*(force_s3tc_enable = getenv("force_s3tc_enable")) &&
+          !strcmp(force_s3tc_enable, "true")*/) {
          debug_printf("couldn't open " DXTN_LIBNAME ", enabling DXTn due to "
             "force_s3tc_enable=true environment variable\n");
          util_format_s3tc_enabled = TRUE;
