From 8156e70294a911c0dc2c762f925bf544354763df Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 31 Jul 2012 11:48:03 +0200
Subject: [PATCH 19/19] dracut.sh: create relative symlinks for /var/lock and
 /var/run

---
 dracut.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 91e510c..dfe0f08 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -827,8 +827,8 @@ if [[ $kernel_only != yes ]]; then
         fi
     done
 
-    ln -sfn /run "$initdir/var/run"
-    ln -sfn /run/lock "$initdir/var/lock"
+    ln -sfn ../run "$initdir/var/run"
+    ln -sfn ../run/lock "$initdir/var/lock"
 else
     for d in lib "$libdir"; do
         [[ -e "${initdir}${prefix}/$d" ]] && continue
-- 
1.7.8.6

