From 5ee9d6bad16c18d17cea4a9ca737bc5583209fa0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Thu, 26 Jul 2012 19:04:30 +0200
Subject: [PATCH 15/15] gentoo.conf: enable ro_mnt

---
 dracut.conf.d/gentoo.conf.example |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/dracut.conf.d/gentoo.conf.example b/dracut.conf.d/gentoo.conf.example
index 411f9a0..6b74954 100644
--- a/dracut.conf.d/gentoo.conf.example
+++ b/dracut.conf.d/gentoo.conf.example
@@ -2,6 +2,7 @@
 # Dracut config file customized for Gentoo Base System release 2
 
 udevdir=/lib/udev
+ro_mnt=yes
 
 #
 # Modules
-- 
1.7.8.6

