From d5ed8b1f8b1240f31953a45992cbc92ee23601bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Thu, 26 Jul 2012 14:54:19 +0200
Subject: [PATCH 10/14] 98usrmount: print mount options

---
 modules.d/98usrmount/mount-usr.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules.d/98usrmount/mount-usr.sh b/modules.d/98usrmount/mount-usr.sh
index 98e08e3..f4d59b3 100755
--- a/modules.d/98usrmount/mount-usr.sh
+++ b/modules.d/98usrmount/mount-usr.sh
@@ -81,7 +81,7 @@ mount_usr()
             fi
         fi
 
-        info "Mounting /usr"
+        info "Mounting /usr with -o $_opts"
         mount "$NEWROOT/usr" 2>&1 | vinfo
 
         if ! ismounted "$NEWROOT/usr"; then
-- 
1.7.8.6

