From d49d600c12ed2a2ed057b082a46e3139ebe6afe5 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 3 Jan 2013 09:29:05 +0100
Subject: [PATCH 2/4] dracut-functions.sh: fixed inst_rules search path

---
 dracut-functions.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index f1c797c..5481368 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -900,7 +900,7 @@ inst_rules() {
     inst_dir "${udevdir}/rules.d"
     inst_dir "$_target"
     for _rule in "$@"; do
-        if [ "${rule#/}" = "$rule" ]; then
+        if [ "${_rule#/}" = "$_rule" ]; then
             for r in ${udevdir}/rules.d /etc/udev/rules.d; do
                 if [[ -f $r/$_rule ]]; then
                     _found="$r/$_rule"
-- 
1.8.0.2

