From bf7a572d8f978e16d92385ebf36e2f837fc2f3d1 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 11 Mar 2013 15:46:37 +0100
Subject: [PATCH 4/5] lsinitrd.sh: removed trailing "}"

---
 lsinitrd.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lsinitrd.sh b/lsinitrd.sh
index 4b8a7e2..5cd8c2a 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -60,7 +60,7 @@ else
     if [[ $MACHINE_ID ]] && ( [[ -d /boot/${MACHINE_ID} ]] || [[ -L /boot/${MACHINE_ID} ]] ); then
         image="/boot/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
     else
-        image="/boot/initramfs-${KERNEL_VERSION}.img}"
+        image="/boot/initramfs-${KERNEL_VERSION}.img"
     fi
 fi
 
-- 
1.8.1.4

