--- ssl.c.orig	2012-04-02 14:49:21.093698637 -0700
+++ ssl.c	2012-04-02 14:48:50.663277244 -0700
@@ -3912,11 +3912,14 @@
       struct env_set *es = session->opt->es;
       struct env_item *e;
       struct buffer out = alloc_buf_gc (512*3, &gc);
+      const char *iv_plat, *iv_plat_rel;
 
       /* push version */
       buf_printf (&out, "IV_VER=%s\n", PACKAGE_VERSION);
 
       /* push platform */
+      iv_plat = getenv("IV_PLAT");
+      if (iv_plat == NULL) {
 #if defined(TARGET_LINUX)
       buf_printf (&out, "IV_PLAT=linux\n");
 #elif defined(TARGET_SOLARIS)
@@ -3932,6 +3935,12 @@
 #elif defined(WIN32)
       buf_printf (&out, "IV_PLAT=win\n");
 #endif
+      } else
+        buf_printf (&out, "IV_PLAT=%s\n", iv_plat);
+
+      iv_plat_rel = getenv("IV_PLAT_REL");
+      if (iv_plat_rel != NULL)
+        buf_printf (&out, "IV_PLAT_REL=%s\n", iv_plat_rel);
 
       /* push mac addr */
       {
