--- apps/openssl.cnf
+++ apps/openssl.cnf
@@ -12,6 +12,21 @@
 #oid_file		= $ENV::HOME/.oid
 oid_section		= new_oids
 
+openssl_conf            = openssl_def
+
+[openssl_def]
+engines = engine_section
+
+[engine_section]
+pkcs11 = pkcs11_section
+
+[pkcs11_section]
+engine_id = pkcs11
+dynamic_path = @GENTOO_LIBDIR@/engines/engine_pkcs11.so
+MODULE_PATH = @GENTOO_LIBDIR@/libchaps.so
+PIN = 111111
+init = 0
+
 # To use this configuration file with the "-extfile" option of the
 # "openssl x509" utility, name here the section containing the
 # X.509v3 extensions to use:
