Fix from upstream cvs

Index: ssl/t1_lib.c
===================================================================
RCS file: /usr/local/src/openssl/CVSROOT/openssl/ssl/t1_lib.c,v
retrieving revision 1.64
retrieving revision 1.65
diff -u -p -r1.64 -r1.65
--- ssl/t1_lib.c	29 Dec 2008 16:15:27 -0000  1.64
+++ ssl/t1_lib.c	28 Apr 2009 22:10:54 -0000  1.65
@@ -267,6 +267,10 @@ unsigned char *ssl_add_clienthello_tlsex
 	int extdatalen=0;
 	unsigned char *ret = p;
 
+	/* don't add extensions for SSLv3 */
+	if (s->client_version == SSL3_VERSION)
+		return p;
+
 	ret+=2;
 
 	if (ret>=limit) return NULL; /* this really never occurs, but ... */
@@ -448,6 +452,10 @@ unsigned char *ssl_add_serverhello_tlsex
 	int extdatalen=0;
 	unsigned char *ret = p;
 
+	/* don't add extensions for SSLv3 */
+	if (s->version == SSL3_VERSION)
+		return p;
+	
 	ret+=2;
 	if (ret>=limit) return NULL; /* this really never occurs, but ... */
