http://rt.openssl.org/Ticket/Display.html?id=2081

--- Configure
+++ Configure
@@ -979,7 +979,8 @@
 my $shared_cflag = $fields[$idx_shared_cflag];
 my $shared_ldflag = $fields[$idx_shared_ldflag];
 my $shared_extension = $fields[$idx_shared_extension];
-my $ranlib = $fields[$idx_ranlib];
+my $ar = $ENV{'AR'} || "ar";
+my $ranlib = $ENV{'RANLIB'} || $fields[$idx_ranlib];
 my $arflags = $fields[$idx_arflags];
 
 if ($fips)
@@ -1487,6 +1488,7 @@
 	s/^RMD160_ASM_OBJ=.*$/RMD160_ASM_OBJ= $rmd160_obj/;
 	s/^PROCESSOR=.*/PROCESSOR= $processor/;
 	s/^RANLIB=.*/RANLIB= $ranlib/;
+	s/^AR=ar /AR= $ar /;
 	s/^ARFLAGS=.*/ARFLAGS= $arflags/;
 	s/^PERL=.*/PERL= $perl/;
 	s/^KRB5_INCLUDES=.*/KRB5_INCLUDES=$withargs{"krb5-include"}/;
