From d5629f76e1775550692be0d94c8e7f82a96f657d Mon Sep 17 00:00:00 2001
From: Nick Owens <mischief@offblast.org>
Date: Thu, 17 Sep 2015 15:27:14 -0700
Subject: [PATCH] remove kernel version check

---
 usr/Makefile | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/usr/Makefile b/usr/Makefile
index 5ac0726..9c436df 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -2,24 +2,8 @@
 
 OSNAME=$(shell uname -s)
 
-# allow users to override these
-# eg to compile for a kernel that you aren't currently running
-KERNELRELEASE ?= $(shell uname -r)
-KSRC ?= /lib/modules/$(KERNELRELEASE)/build
-
-KSUBLEVEL=$(shell cat $(KSRC)/Makefile | awk -F= '/^SUBLEVEL =/ {print $$2}' | \
-			sed 's/^[ \t]*//;s/[ \t]*$$//')
-
 ifeq ($(OSNAME),Linux)
-	ifeq ($(KSUBLEVEL),11)
-		IPC_CFLAGS=-DNETLINK_ISCSI=12 -D_GNU_SOURCE
-	else
-	ifeq ($(KSUBLEVEL),12)
-		IPC_CFLAGS=-DNETLINK_ISCSI=12 -D_GNU_SOURCE
-	else
-		IPC_CFLAGS=-DNETLINK_ISCSI=8 -D_GNU_SOURCE
-	endif
-	endif
+IPC_CFLAGS=-DNETLINK_ISCSI=8 -D_GNU_SOURCE
 IPC_OBJ=netlink.o
 else
 ifeq ($(OSNAME),FreeBSD)
-- 
2.4.6

