--- Makefile.am.orig	2010-05-24 11:26:37.000000000 -0700
+++ Makefile.am	2010-05-24 11:31:05.000000000 -0700
@@ -299,6 +299,8 @@
 # @SOCKET_LIBS@ or @NSL_LIBS@, as those should also be included in
 # @GTK_LIBS@ (as those are also needed for X applications, and GTK+
 # applications are X applications).
+wireshark_DEPENDENCIES = \
+	@INET_PTON_LO@
 wireshark_LDADD = \
 	$(wireshark_optional_objects)	\
 	gtk/libui.a			\
@@ -330,6 +332,8 @@
 endif
 
 # Libraries and plugin flags with which to link tshark.
+tshark_DEPENDENCIES = \
+	@INET_PTON_LO@
 tshark_LDADD = \
 	$(wireshark_optional_objects)	\
 	wiretap/libwiretap.la		\
@@ -360,6 +364,8 @@
 endif
 
 # Libraries and plugin flags with which to link tshark.
+rawshark_DEPENDENCIES = \
+	@INET_PTON_LO@
 rawshark_LDADD = \
 	$(wireshark_optional_objects)	\
 	wiretap/libwiretap.la		\
