diff -ru vim73.orig/src/link.sh vim73/src/link.sh
--- vim73.orig/src/link.sh	2010-10-21 16:29:07 +0200
+++ vim73/src/link.sh	2010-10-21 16:23:15 +0200
@@ -41,7 +41,7 @@
   if sh link.cmd; then
     touch auto/link.sed
     cp link.cmd linkit.sh
-    for libname in SM ICE nsl dnet dnet_stub inet socket dir elf iconv Xt Xmu Xp Xpm X11 Xdmcp x w perl dl pthread thread readline m crypt attr; do
+    for libname in dummy; do
       cont=yes
       while test -n "$cont"; do
         if grep "l$libname " linkit.sh >/dev/null; then
