https://bugs.gentoo.org/show_bug.cgi?id=520112
diff --git a/pym/_emerge/depgraph.py b/pym/_emerge/depgraph.py
index a10297a..2d30966 100644
--- a/pym/_emerge/depgraph.py
+++ b/pym/_emerge/depgraph.py
@@ -3496,13 +3496,13 @@ class depgraph(object):
 		a favorite list."""
 		debug = "--debug" in self._frozen_config.myopts
 		onlydeps = "--onlydeps" in self._frozen_config.myopts
-		myroot = self._frozen_config.target_root
-		pkgsettings = self._frozen_config.pkgsettings[myroot]
-		pprovideddict = pkgsettings.pprovideddict
-		virtuals = pkgsettings.getvirtuals()
 		args = self._dynamic_config._initial_arg_list[:]
 
 		for arg in self._expand_set_args(args, add_to_digraph=True):
+			myroot = arg.root_config.root
+			pkgsettings = self._frozen_config.pkgsettings[myroot]
+			pprovideddict = pkgsettings.pprovideddict
+			virtuals = pkgsettings.getvirtuals()
 			for atom in arg.pset.getAtoms():
 				self._spinner_update()
 				dep = Dependency(atom=atom, onlydeps=onlydeps,
