From 1d1338afa6aa555c5f6c83d07fceec43a4d87f0d Mon Sep 17 00:00:00 2001
From: Sergey V. Udaltsov <svu@gnome.org>
Date: Wed, 05 Oct 2011 21:26:26 +0000
Subject: Levels 2-4 for CTRL+ALT are propagated from level 1

---
diff --git a/symbols/keypad b/symbols/keypad
index e85aaac..1bab391 100644
--- a/symbols/keypad
+++ b/symbols/keypad
@@ -84,19 +84,19 @@ xkb_symbols "x11" {
 // Ungrab cancels server/keyboard/pointer grabs
     key <KPDV> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Divide, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Ungrab ]
+        symbols[Group1]= [ KP_Divide, KP_Divide, KP_Divide, KP_Divide, XF86_Ungrab ]
     };
 
 // ClsGrb kills whichever client has a grab in effect
     key <KPMU> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Multiply,	VoidSymbol, VoidSymbol, VoidSymbol, XF86_ClearGrab ]
+        symbols[Group1]= [ KP_Multiply,	KP_Multiply, KP_Multiply, KP_Multiply, XF86_ClearGrab ]
     };
 
 // -VMode switches to the previous video mode
     key <KPSU> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Subtract, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Prev_VMode ]
+        symbols[Group1]= [ KP_Subtract, KP_Subtract, KP_Subtract, KP_Subtract, XF86_Prev_VMode ]
     };
 
     key  <KP7> {	[  KP_Home,	KP_7	]	};
@@ -106,7 +106,7 @@ xkb_symbols "x11" {
 // +VMode switches to the next video mode
     key <KPAD> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Add, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Next_VMode ]
+        symbols[Group1]= [ KP_Add, KP_Add, KP_Add, KP_Add, XF86_Next_VMode ]
     };
 
     key  <KP4> {	[  KP_Left,	KP_4	]	};
@@ -242,11 +242,11 @@ xkb_symbols "legacymath" {
 
     key.type[Group1]="CTRL+ALT" ;
 
-    key <KPDV> { [   KP_Divide, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Ungrab ] }; // / <XF86_Ungrab>
-    key <KPMU> { [ KP_Multiply, VoidSymbol, VoidSymbol, VoidSymbol, XF86_ClearGrab ] }; // * <XF86_ClearGrab>
-    key <KPSU> { [ KP_Subtract, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Prev_VMode ] }; // - <XF86_Prev_VMode>
+    key <KPDV> { [   KP_Divide, KP_Divide, KP_Divide, KP_Divide, XF86_Ungrab ] }; // / <XF86_Ungrab>
+    key <KPMU> { [ KP_Multiply, KP_Multiply, KP_Multiply, KP_Multiply, XF86_ClearGrab ] }; // * <XF86_ClearGrab>
+    key <KPSU> { [ KP_Subtract, KP_Subtract, KP_Subtract, KP_Subtract, XF86_Prev_VMode ] }; // - <XF86_Prev_VMode>
 
-    key <KPAD> { [      KP_Add, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Next_VMode ] }; // + <XF86_Next_VMode>
+    key <KPAD> { [      KP_Add, KP_Add, KP_Add, KP_Add, XF86_Next_VMode ] }; // + <XF86_Next_VMode>
 
 };
 
@@ -602,7 +602,7 @@ xkb_symbols "numoperdecsep" {
 // ClsGrb kills whichever client has a grab in effect
     key <KOMU> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Multiply, VoidSymbol, VoidSymbol, VoidSymbol, XF86_ClearGrab ]
+        symbols[Group1]= [ KP_Multiply, KP_Multiply, KP_Multiply, KP_Multiply, XF86_ClearGrab ]
     };
 
     key  <KO4> {	[  KP_4	]	};
@@ -611,7 +611,7 @@ xkb_symbols "numoperdecsep" {
 // -VMode switches to the previous video mode
     key <KOSU> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Subtract, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Prev_VMode ]
+        symbols[Group1]= [ KP_Subtract, KP_Subtract, KP_Subtract, KP_Subtract, XF86_Prev_VMode ]
     };
 
     key  <KO1> {	[  KP_1	]	};
@@ -620,7 +620,7 @@ xkb_symbols "numoperdecsep" {
 // +VMode switches to the next video mode
     key <KOAD> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Add, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Next_VMode ]
+        symbols[Group1]= [ KP_Add, KP_Add, KP_Add, KP_Add, XF86_Next_VMode ]
     };
 
     key  <KO0> {	[  KP_0	]	};
@@ -628,6 +628,6 @@ xkb_symbols "numoperdecsep" {
 // Ungrab cancels server/keyboard/pointer grabs
     key <KODV> {
         type="CTRL+ALT",
-        symbols[Group1]= [ KP_Divide, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Ungrab ]
+        symbols[Group1]= [ KP_Divide, KP_Divide, KP_Divide, KP_Divide, XF86_Ungrab ]
     };
 };
diff --git a/symbols/srvr_ctrl b/symbols/srvr_ctrl
index 7d47d66..73b5af2 100644
--- a/symbols/srvr_ctrl
+++ b/symbols/srvr_ctrl
@@ -12,25 +12,25 @@ xkb_symbols "stdkeypad" {
 // Ungrab cancels server/keyboard/pointer grabs
     key <KPDV> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ KP_Divide, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Ungrab ]
+	symbols[Group1]= [ KP_Divide, KP_Divide, KP_Divide, KP_Divide, XF86_Ungrab ]
     };
 
 // ClsGrb kills whichever client has a grab in effect
     key <KPMU> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ KP_Multiply, VoidSymbol, VoidSymbol, VoidSymbol, XF86_ClearGrab ]
+	symbols[Group1]= [ KP_Multiply, KP_Multiply, KP_Multiply, KP_Multiply, XF86_ClearGrab ]
     };
 
 // -VMode switches to the previous video mode
     key <KPSU> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ KP_Subtract, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Prev_VMode ]
+	symbols[Group1]= [ KP_Subtract, KP_Subtract, KP_Subtract, KP_Subtract, XF86_Prev_VMode ]
     };
 
 // +VMode switches to the next video mode
     key <KPAD> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ KP_Add, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Next_VMode]
+	symbols[Group1]= [ KP_Add, KP_Add, KP_Add, KP_Add, XF86_Next_VMode]
     };
 
 };
@@ -40,62 +40,62 @@ xkb_symbols "fkey2vt" {
 
     key <FK01> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F1, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_1 ]
+	symbols[Group1]= [ F1, F1, F1, F1, XF86_Switch_VT_1 ]
     };
 
     key <FK02> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F2, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_2 ]
+	symbols[Group1]= [ F2, F2, F2, F2, XF86_Switch_VT_2 ]
     };
 
     key <FK03> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F3, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_3 ]
+	symbols[Group1]= [ F3, F3, F3, F3, F3 ]
     };
 
     key <FK04> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F4, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_4 ]
+	symbols[Group1]= [ F4, F4, F4, F4, F4 ]
     };
 
     key <FK05> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F5, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_5 ]
+	symbols[Group1]= [ F5, F5, F5, F5, F5 ]
     };
 
     key <FK06> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F6, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_6 ]
+	symbols[Group1]= [ F6, F6, F6, F6, F6 ]
     };
 
     key <FK07> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F7, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_7 ]
+	symbols[Group1]= [ F7, F7, F7, F7, F7 ]
     };
 
     key <FK08> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F8, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_8 ]
+	symbols[Group1]= [ F8, F8, F8, F8, F8 ]
     };
 
     key <FK09> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F9, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_9 ]
+	symbols[Group1]= [ F9, F9, F9, F9, F9 ]
     };
 
     key <FK10> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F10, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_10 ]
+	symbols[Group1]= [ F10, F10, F10, F10, F10 ]
     };
 
     key <FK11> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F11, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_11 ]
+	symbols[Group1]= [ F11, F11, F11, F11, F11 ]
     };
 
     key <FK12> {
 	type="CTRL+ALT",
-	symbols[Group1]= [ F12, VoidSymbol, VoidSymbol, VoidSymbol, XF86_Switch_VT_12 ]
+	symbols[Group1]= [ F12, F12, F12, F12, F12 ]
     };
 
 };
diff --git a/symbols/terminate b/symbols/terminate
index 96dd6e8..c74220b 100644
--- a/symbols/terminate
+++ b/symbols/terminate
@@ -2,6 +2,6 @@ partial default modifier_keys
 xkb_symbols "ctrl_alt_bksp" {
     key <BKSP> {
         type="CTRL+ALT",
-        symbols[Group1] = [ NoSymbol, VoidSymbol, VoidSymbol, VoidSymbol, Terminate_Server ]
+        symbols[Group1] = [ NoSymbol, NoSymbol, NoSymbol, NoSymbol, Terminate_Server ]
     };
 };
--
cgit v0.9.0.2-2-gbebe
