diff -urN xkeyboard-config-2.4.1.orig/rules/compat/layoutsMapping.lst xkeyboard-config-2.4.1/rules/compat/layoutsMapping.lst
--- xkeyboard-config-2.4.1.orig/rules/compat/layoutsMapping.lst	2012-11-14 14:08:33.999093026 +0900
+++ xkeyboard-config-2.4.1/rules/compat/layoutsMapping.lst	2012-11-14 17:03:26.637971159 +0900
@@ -25,3 +25,6 @@
 yu	srp
 fr-latin9 fr(latin9)
 us_intl	us(alt-intl)
+# To be us(euro) as the default us keyboard layout on parrot, overwrite us
+# layout mapping to us(euro).
+us us(euro)
