diff -ur xkeyboard-config-2.4.1.orig/rules/base.xml.in xkeyboard-config-2.4.1/rules/base.xml.in
--- xkeyboard-config-2.4.1.orig/rules/base.xml.in	2011-10-05 07:19:54.000000000 +0900
+++ xkeyboard-config-2.4.1/rules/base.xml.in	2012-02-14 19:20:13.944976247 +0900
@@ -3708,7 +3708,8 @@
         <_shortDescription>no</_shortDescription>
         <_description>Norwegian</_description>
         <languageList>
-          <iso639Id>nor</iso639Id>
+          <iso639Id>nob</iso639Id>
+          <iso639Id>nno</iso639Id>
         </languageList>
       </configItem>
       <variantList>
