diff -urN xkeyboard-config-2.4.1.org/symbols/pc xkeyboard-config-2.4.1/symbols/pc
--- xkeyboard-config-2.4.1.org/symbols/pc	2013-01-17 13:51:15.974219536 +0900
+++ xkeyboard-config-2.4.1/symbols/pc	2013-01-17 14:31:45.361168332 +0900
@@ -42,7 +42,9 @@
     modifier_map Shift  { Shift_L, Shift_R };
     modifier_map Lock   { Caps_Lock, ISO_Lock };
     modifier_map Control{ Control_L, Control_R };
-    modifier_map Mod2   { Num_Lock };
+    // Keep Num_Lock as Mod2 modifier. Mod2 is used as both NumLock modifier
+    // and diamond key modifier.
+    modifier_map Mod2   { Num_Lock, XF86Launch6 };
     // Use <CAPS> as Mod3. The Mod3Mask mask will be remapped to ControlMask,
     // Mod1Mask (Alt), etc. in Chrome. crbug.com/146204
     modifier_map Mod3   { <CAPS> };
