Index: configure.ac
===================================================================
RCS file: /cvsroot/ddccontrol/ddccontrol/configure.ac,v
retrieving revision 1.40
diff -u -r1.40 configure.ac
--- configure.ac	26 Jul 2006 22:02:15 -0000	1.40
+++ configure.ac	1 Mar 2007 14:49:35 -0000
@@ -101,7 +101,10 @@
 DDCPCI=
 if test x$support_ddcpci = xyes; then
    AC_CHECK_HEADERS([pci/pci.h], [], [AC_MSG_ERROR([PCI utils headers not found, please install pci-utils.], [1])], [])
-   AC_CHECK_LIB([pci], [pci_alloc], [], [AC_MSG_ERROR([PCI utils library not found, please install pci-utils.], [1])])
+   AC_CHECK_LIB([pci], [pci_alloc], [], [
+      AC_CHECK_LIB([z], [gzopen], [], [AC_MSG_ERROR([PCI utils library not found, please install pci-utils.], [1])])
+      AC_CHECK_LIB([pci], [pci_fill_info], [], [AC_MSG_ERROR([PCI utils library not found, please install pci-utils.], [1])], [-lz])
+   ])
    DDCPCI=ddcpci
    AC_DEFINE_UNQUOTED(HAVE_DDCPCI, 1, [Define if ddccontrol is built with ddcpci support.])
 fi
