diff --git a/SConstruct b/SConstruct
index 11219ed..41d12a6 100644
--- a/SConstruct
+++ b/SConstruct
@@ -31,7 +31,7 @@ def common_env():
     env['LINKFLAGS'] += SCons.Util.CLVar(os.environ['LDFLAGS'])
 
   # Fix issue with scons not passing some vars through the environment.
-  for key in Split('PKG_CONFIG_LIBDIR PKG_CONFIG_PATH SYSROOT'):
+  for key in Split('PATH PKG_CONFIG_LIBDIR PKG_CONFIG_PATH SYSROOT'):
     if os.environ.has_key(key):
       env['ENV'][key] = os.environ[key]
 
