diff -Naur protobuf-2.3.0-orig/python/setup.py protobuf-2.3.0/python/setup.py
--- protobuf-2.3.0-orig/python/setup.py	2011-03-22 16:33:33.000000000 -0700
+++ protobuf-2.3.0/python/setup.py	2011-03-22 17:01:14.000000000 -0700
@@ -16,7 +16,9 @@
 maintainer_email = "protobuf@googlegroups.com"
 
 # Find the Protocol Compiler.
-if os.path.exists("../src/protoc"):
+if 'PROTOC' in os.environ and os.path.exists(os.environ['PROTOC']):
+  protoc = os.environ['PROTOC']
+elif os.path.exists("../src/protoc"):
   protoc = "../src/protoc"
 elif os.path.exists("../src/protoc.exe"):
   protoc = "../src/protoc.exe"
