diff --git a/common.mk b/common.mk
index 058e906..9d4158a 100644
--- a/common.mk
+++ b/common.mk
@@ -276,7 +276,7 @@ endif
 #  CXXFLAGS := -mahflag $(CXXFLAGS) # Prepend to the list
 #  CXXFLAGS := $(filter-out badflag,$(CXXFLAGS)) # Filter out a value
 # The same goes for CFLAGS.
-COMMON_CFLAGS-gcc := -fstack-protector-strong -fvisibility=internal -ggdb3 \
+COMMON_CFLAGS-gcc := -fstack-protector-all -fvisibility=internal -ggdb3 \
   -Wa,--noexecstack
 COMMON_CFLAGS-clang := -fstack-protector-all -fvisibility=hidden -ggdb
 COMMON_CFLAGS := -Wall -Werror -fno-strict-aliasing -O1 -Wformat=2
