--- a/rules/rules.d/50-udev-default.rules	2013-01-23 16:05:36.000000000 -0500
+++ b/rules/rules.d/50-udev-default.rules	2013-01-23 16:05:50.000000000 -0500
@@ -99,7 +99,7 @@
 KERNEL=="cpu[0-9]*",		MODE="0444"
 
 KERNEL=="fuse",			MODE="0666", OPTIONS+="static_node=fuse", \
-  RUN+="/bin/mount -t fusectl fusectl /sys/fs/fuse/connections"
+  RUN+="/bin/mount -t fusectl -o nodev,noexec,nosuid fusectl /sys/fs/fuse/connections"
 
 SUBSYSTEM=="rtc", DRIVERS=="rtc_cmos", SYMLINK+="rtc"
 KERNEL=="mmtimer",		MODE="0644"
