diff -aur llvm-3.1.src.orig/configure llvm-3.1.src/configure
--- llvm-3.1.src.orig/configure	2012-05-11 13:48:57.000000000 -0700
+++ llvm-3.1.src/configure	2012-10-08 14:34:48.591805189 -0700
@@ -3486,8 +3486,6 @@
 do
   if test -d ${srcdir}/projects/${i} ; then
     case ${i} in
-      sample)       subdirs="$subdirs projects/sample"
-    ;;
       privbracket)  subdirs="$subdirs projects/privbracket"
  ;;
       llvm-stacker) subdirs="$subdirs projects/llvm-stacker"
diff -aur llvm-3.1.src.orig/projects/Makefile llvm-3.1.src/projects/Makefile
--- llvm-3.1.src.orig/projects/Makefile	2010-09-09 08:49:32.000000000 -0700
+++ llvm-3.1.src/projects/Makefile	2012-10-08 14:51:55.783370351 -0700
@@ -24,5 +24,6 @@
 ifeq ($(ARCH), Sparc)
 DIRS := $(filter-out sample, $(DIRS))
 endif
+DIRS := $(filter-out sample, $(DIRS))
 
 include $(PROJ_SRC_ROOT)/Makefile.rules
