diff -aur llvm-3.1.src.orig/configure llvm-3.1.src/configure
--- llvm-3.1.src.orig/configure	2012-05-11 13:48:57.000000000 -0700
+++ llvm-3.1.src/configure	2012-10-08 13:38:31.363732213 -0700
@@ -1952,13 +1952,13 @@
 
 
 
-if test ${srcdir} != "." ; then
-  if test -f ${srcdir}/include/llvm/Config/config.h ; then
-    { { echo "$as_me:$LINENO: error: Already configured in ${srcdir}" >&5
-echo "$as_me: error: Already configured in ${srcdir}" >&2;}
-   { (exit 1); exit 1; }; }
-  fi
-fi
+#if test ${srcdir} != "." ; then
+#  if test -f ${srcdir}/include/llvm/Config/config.h ; then
+#    { { echo "$as_me:$LINENO: error: Already configured in ${srcdir}" >&5
+#echo "$as_me: error: Already configured in ${srcdir}" >&2;}
+#   { (exit 1); exit 1; }; }
+#  fi
+#fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
@@ -4798,7 +4798,7 @@
   test -z "$BUILD_CC" && { { echo "$as_me:$LINENO: error: no acceptable cc found in \$PATH" >&5
 echo "$as_me: error: no acceptable cc found in \$PATH" >&2;}
    { (exit 1); exit 1; }; }
-  ac_build_link='${BUILD_CC-cc} -o conftest $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+  ac_build_link='${BUILD_CC-cc} -o conftest $HOST_CFLAGS $HOST_CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
   rm -f conftest*
   echo 'int main () { return 0; }' > conftest.$ac_ext
   ac_cv_build_exeext=
Only in llvm-3.1.src/: configure.orig
