diff -aur llvm-3.1.src.orig/Makefile llvm-3.1.src/Makefile
--- llvm-3.1.src.orig/Makefile	2012-01-16 18:56:49.000000000 -0800
+++ llvm-3.1.src/Makefile	2012-10-08 13:39:18.854266324 -0700
@@ -114,6 +114,11 @@
 	  unset CXXFLAGS ; \
 	  unset SDKROOT ; \
 	  unset UNIVERSAL_SDK_PATH ; \
+	  AR=$(HOST_AR) ;\
+	  AS=$(HOST_AS) ;\
+	  LD=$(HOST_LD) ;\
+	  CC=$(HOST_CC) ;\
+	  CXX=$(HOST_CXX) ;\
 	  $(PROJ_SRC_DIR)/configure --build=$(BUILD_TRIPLE) \
 		--host=$(BUILD_TRIPLE) --target=$(BUILD_TRIPLE) \
 	        --disable-polly ; \
Only in llvm-3.1.src/: Makefile.orig
