--- ./glibc-2.11.1/csu/Makefile	2012-02-15 19:52:31.597382895 -0800
+++ ./glibc-2.11.1/csu/Makefile	2012-02-15 19:53:24.557242505 -0800
@@ -93,7 +93,7 @@
 $(crtstuff:%=$(objpfx)%.o): %.o: %.S $(objpfx)defs.h
        $(compile.S) -g0 $(ASFLAGS-.os) -o $@
 
-CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions $(fno-unit-at-a-time)
+CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions $(fno-unit-at-a-time) -fno-record-gcc-switches
 
 vpath initfini.c $(sysdirs)
