From b75e17459bf1201c40e1ff372ec43c1c33caa57a Mon Sep 17 00:00:00 2001
From: Alex Crawford <alex.crawford@coreos.com>
Date: Wed, 18 May 2016 18:45:18 -0700
Subject: [PATCH] Hardcode the root directory

---
 bin/gcloud | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/gcloud b/bin/gcloud
index e3bec46..ef62c98 100755
--- a/bin/gcloud
+++ b/bin/gcloud
@@ -58,7 +58,7 @@ _cloudsdk_root_dir() {
       esac
   done
 }
-CLOUDSDK_ROOT_DIR=$(_cloudsdk_root_dir "$0")
+CLOUDSDK_ROOT_DIR="/usr/lib/google-cloud-sdk"
 
 # Cloud SDK requires python 2 (2.6 or 2.7)
 case $CLOUDSDK_PYTHON in
-- 
2.7.3

