diff -urN ../mozc-1.4.1033.102.orig/unix/ibus/mozc_engine.cc ./unix/ibus/mozc_engine.cc
--- ../mozc-1.4.1033.102.orig/unix/ibus/mozc_engine.cc	2012-04-05 18:26:30.601368015 +0900
+++ ./unix/ibus/mozc_engine.cc	2012-04-05 19:31:20.549084704 +0900
@@ -225,15 +225,16 @@
   DCHECK(delta);
 
   COMPILE_ASSERT(sizeof(int64) > sizeof(guint), int64_guint_check);
+  COMPILE_ASSERT(sizeof(int64) == sizeof(llabs(0)), int64_llabs_check);
   const int64 kInt32AbsMax =
-      abs(static_cast<int64>(numeric_limits<int32>::max()));
+      llabs(static_cast<int64>(numeric_limits<int32>::max()));
   const int64 kInt32AbsMin =
-      abs(static_cast<int64>(numeric_limits<int32>::min()));
+      llabs(static_cast<int64>(numeric_limits<int32>::min()));
   const int64 kInt32SafeAbsMax =
       min(kInt32AbsMax, kInt32AbsMin);
 
   const int64 diff = static_cast<int64>(from) - static_cast<int64>(to);
-  if (abs(diff) > kInt32SafeAbsMax) {
+  if (llabs(diff) > kInt32SafeAbsMax) {
     return false;
   }
 
