diff -urN ../mozc-1.4.1033.102.orig/languages/hangul/session.cc ./languages/hangul/session.cc
--- ../mozc-1.4.1033.102.orig/languages/hangul/session.cc	2012-09-21 16:45:37.703125368 +0900
+++ ./languages/hangul/session.cc	2012-09-21 16:57:41.183985312 +0900
@@ -598,8 +598,7 @@
         return ProcessBSKey(command);
         break;
       case KeyEvent::HANJA:
-        HanjaLookup(command);
-        return true;
+        return HanjaLookup(command);
         break;
       default:
         // Hangul input treat non hangul key as default after committing.
Binary files ../mozc-1.4.1033.102.orig/languages/hangul/.session.cc.swp and ./languages/hangul/.session.cc.swp differ
