--- base/time_posix.cc.orig	2012-05-30 10:06:00.571692591 -0700
+++ base/time_posix.cc	2012-05-30 10:06:25.911766484 -0700
@@ -24,7 +24,7 @@ struct timespec TimeDelta::ToTimeSpec()
   }
   struct timespec result =
       {seconds,
-       microseconds * Time::kNanosecondsPerMicrosecond};
+       static_cast<long>(microseconds * Time::kNanosecondsPerMicrosecond)};
   return result;
 }
 
--- base/message_pump_libevent.cc.orig	2012-05-30 10:04:00.331341859 -0700
+++ base/message_pump_libevent.cc	2012-05-30 10:04:25.781416110 -0700
@@ -6,6 +6,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "base/auto_reset.h"
 #include "base/compiler_specific.h"
--- o3d/core/cross/param.h.orig	2012-05-30 10:02:52.981145332 -0700
+++ o3d/core/cross/param.h	2012-05-30 10:03:31.441257564 -0700
@@ -444,7 +444,7 @@ class TypedParam : public TypedParamBase
 
   // Copies the data from another Param.
   virtual void CopyDataFromParam(Param* source_param) {
-    set_value_private((down_cast<TypedParam*>(source_param))->value_);
+    this->set_value_private((down_cast<TypedParam*>(source_param))->value_);
   }
 
  private:
@@ -552,7 +552,7 @@ class TypedRefParam : public TypedRefPar
   // Copies the data from another Param.
   virtual void CopyDataFromParam(Param* source_param) {
     T* value = down_cast<TypedRefParam*>(source_param)->value_.Get();
-    set_value_private(value ? value->GetWeakPointer() : WeakPointerType());
+    this->set_value_private(value ? value->GetWeakPointer() : WeakPointerType());
   }
 
  private:
