From dfbfc78a1d6ecae7583fdb8b681276f6a8f7ca8a Mon Sep 17 00:00:00 2001
From: Camilo Aguilar <camilo.aguilar@gmail.com>
Date: Thu, 2 Oct 2014 22:00:38 -0400
Subject: [PATCH] Adds --with-libdnet=DIR option

---
 open-vm-tools/configure.ac | 78 +++++++++++++++++++++-------------------------
 1 file changed, 35 insertions(+), 43 deletions(-)

diff --git a/open-vm-tools/configure.ac b/open-vm-tools/configure.ac
index 6d52905..cbaf51c 100644
--- a/open-vm-tools/configure.ac
+++ b/open-vm-tools/configure.ac
@@ -666,49 +666,41 @@ if test "$with_procps" != "yes"; then
 AC_DEFINE([NO_PROCPS], 1, [Define to 1 if building without procps.])
 fi
 
-AC_ARG_WITH([dnet],
-	    [AS_HELP_STRING([--without-dnet],
-	    [compiles without libdnet (disables support for nicinfo)])],
-	    [],
-	    [with_dnet=yes])
-
-have_dnet="no"
-if test "$with_dnet" = "yes"; then
-	# On Debian, dnet is installed via the libdumbnet package. We need to
-	# detect this so that our source files include dumbnet.h instead of
-	# dnet.h, which is part of a different package altogether.
-   AC_VMW_CHECK_LIB([dumbnet],
-                    [DNET],
-                    [],
-                    [dumbnet-config],
-                    [],
-                    [dumbnet.h],
-                    [intf_open],
-                    [have_dnet="yes";
-                     AC_DEFINE([DNET_IS_DUMBNET], 1, [Define to 1 if substituting Debian's libdumbnet for libdnet.])],
-                    [])
-
-   if test $have_dnet = "no"; then
-      AC_VMW_CHECK_LIB([dnet],
-                       [DNET],
-                       [],
-                       [dnet-config],
-                       [],
-                       [dnet.h],
-                       [intf_open],
-                       [have_dnet="yes"],
-                       [])
-   fi
-
-   if test $have_dnet = "no"; then
-		AC_MSG_ERROR(
-		   [dnet-config was not found on your PATH. Please configure without dnet (using --without-dnet) or install dnet - http://libdnet.sourceforge.net])
-	fi
-fi
-
-if test "$with_dnet" != "yes"; then
-AC_DEFINE([NO_DNET], 1, [Define to 1 if building without libdnet.])
-fi
+dnl Checks for (installed) libdnet
+AC_ARG_WITH(libdnet,
+[  --with-libdnet=DIR      use libdnet in DIR],
+[ case "$withval" in
+  yes|no)
+     AC_ERROR([Please specify directory containing dnet-config when using --with-libdnet])
+     ;;
+  *)
+     AC_MSG_CHECKING(for libdnet)
+     AC_MSG_RESULT($withval)
+     if test -f $withval/src/libdnet.a; then
+        DNETINC="-I$withval/include"
+        DNETLIB="-L$withval/src -ldnet `$withval/dnet-config --libs`"
+     elif test -x $withval/bin/dnet-config; then
+        DNETINC="`$withval/bin/dnet-config --cflags`"
+        DNETLIB="`$withval/bin/dnet-config --libs`"
+     else
+        AC_MSG_RESULT(no)
+        AC_ERROR(dnet-config not found in $withval/bin)
+     fi
+     ;;
+  esac
+  AC_MSG_RESULT(yes) ],
+[ dnl This is the default case so let's just use AC_PATH_PROG! --CPK.
+  AC_PATH_PROG(dnetconfig, dnet-config, "no")
+  if test "$dnetconfig" = "no"; then
+     AC_ERROR(dnet-config not found)
+  else
+     DNETINC="`$dnetconfig --cflags`"
+     DNETLIB="`$dnetconfig --libs`"
+  fi]
+)
+
+AC_SUBST(DNETINC)
+AC_SUBST(DNETLIB)
 
 AC_ARG_WITH([icu],
             [AS_HELP_STRING([--without-icu],
-- 
1.9.1

