http://bugs.gentoo.org/332703

--- Makefile
+++ Makefile
@@ -28,10 +28,13 @@
 ping6: LDLIBS += -lidn
 endif
 
+arping: LDLIBS += -lsysfs
+ping6: LDLIBS += -lresolv -lcrypto
+
 tftpd: tftpd.o tftpsubs.o
-arping: arping.o -lsysfs
+arping: arping.o
 ping: ping.o ping_common.o
-ping6: ping6.o ping_common.o -lresolv -lcrypto
+ping6: ping6.o ping_common.o
 ping.o ping6.o ping_common.o: ping_common.h
 tftpd.o tftpsubs.o: tftp.h
 
