diff -Naur gtest-1.4.0.orig/Makefile.am gtest-1.4.0/Makefile.am
--- gtest-1.4.0.orig/Makefile.am	2009-10-02 16:04:30.000000000 +0900
+++ gtest-1.4.0/Makefile.am	2010-01-20 16:52:34.000000000 +0900
@@ -305,7 +305,7 @@
 TESTS += test/gtest-unittest-api_test
 check_PROGRAMS += test/gtest-unittest-api_test
 test_gtest_unittest_api_test_SOURCES = test/gtest-unittest-api_test.cc
-test_gtest_unittest_api_test_LDADD = lib/libgtest_main.la
+test_gtest_unittest_api_test_LDADD = lib/libgtest.la
 
 TESTS += test/gtest-listener_test
 check_PROGRAMS += test/gtest-listener_test
