--- src/Makefile.am	2005-08-01 12:29:59.000000000 +0200
+++ src.az/Makefile.am	2005-08-01 12:30:44.000000000 +0200
@@ -45,6 +45,8 @@ noinst_PROGRAMS = id sulogin
 
 suidbins       = su
 suidubins      = chage chfn chsh expiry gpasswd newgrp passwd
+suidbinperms   = 4711
+suidubinperms  = 4711
 
 LDADD 	       = $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
@@ -79,8 +81,8 @@ install-am: all-am
 	ln -sf newgrp	$(DESTDIR)$(ubindir)/sg
 	ln -sf vipw	$(DESTDIR)$(usbindir)/vigr
 	for i in $(suidbins); do \
-		chmod -f 4755 $(DESTDIR)$(bindir)/$$i; \
+		chmod -f $(suidbinperms) $(DESTDIR)$(bindir)/$$i; \
 	done
 	for i in $(suidubins); do \
-		chmod -f 4755 $(DESTDIR)$(ubindir)/$$i; \
+		chmod -f $(suidubinperms) $(DESTDIR)$(ubindir)/$$i; \
 	done
--- src/Makefile.in	2005-08-01 12:31:07.000000000 +0200
+++ src.az/Makefile.in	2005-08-01 12:33:54.000000000 +0200
@@ -346,6 +346,8 @@ INCLUDES = \
 
 suidbins = su
 suidubins = chage chfn chsh expiry gpasswd newgrp passwd
+suidbinperms = 4711
+suidubinperms = 4711
 LDADD = $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
 
@@ -839,10 +841,10 @@ install-am: all-am
 	ln -sf newgrp	$(DESTDIR)$(ubindir)/sg
 	ln -sf vipw	$(DESTDIR)$(usbindir)/vigr
 	for i in $(suidbins); do \
-		chmod -f 4755 $(DESTDIR)$(bindir)/$$i; \
+		chmod -f $(suidbinperms) $(DESTDIR)$(bindir)/$$i; \
 	done
 	for i in $(suidubins); do \
-		chmod -f 4755 $(DESTDIR)$(ubindir)/$$i; \
+		chmod -f $(suidubinperms) $(DESTDIR)$(ubindir)/$$i; \
 	done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
