--- a/trunk/src/newgrp.c	2008/04/16 22:03:43	1975
+++ b/trunk/src/newgrp.c	2008/04/16 22:04:46	1976
@@ -53,6 +53,10 @@
 static char *Prog;
 static int is_newgrp;
 
+#ifdef WITH_AUDIT
+char audit_buf[80];
+#endif
+
 /* local function prototypes */
 static void usage (void);
 static void check_perms (const struct group *grp,
@@ -349,8 +353,6 @@
 #endif
 
 #ifdef WITH_AUDIT
-	char audit_buf[80];
-
 	audit_help_open ();
 #endif
 	setlocale (LC_ALL, "");
