Index: gtk/gdk.override
===================================================================
--- gtk/gdk.override	(révision 3082)
+++ gtk/gdk.override	(copie de travail)
@@ -44,7 +44,7 @@
 #define GDK_DISPLAY(object) (GDK_DISPLAY_OBJECT(object))
 
 #ifdef HAVE_NUMPY
-#  include <Numeric/arrayobject.h>
+#  include <numpy/arrayobject.h>
 static int have_numpy(void);
 #endif
 
Index: configure.ac
===================================================================
--- configure.ac	(révision 3082)
+++ configure.ac	(copie de travail)
@@ -350,18 +350,27 @@
 esac
 
 
-dnl checks to see if Numeric Python is installed.
+dnl checks to see if numpy is installed.
 AC_ARG_ENABLE(numpy,
   AC_HELP_STRING([--disable-numpy], [Disable numeric python features]),,
   enable_numpy=yes)
 
 if test "x$enable_numpy" != xno; then
   save_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
-  AC_CHECK_HEADER([Numeric/arrayobject.h],
-    [AC_DEFINE(HAVE_NUMPY,,[whether to include numeric python support])],,
-    [#include <Python.h>])
-  CPPFLAGS="$save_CPPFLAGS"
+  numpy_INCLUDES=`$PYTHON -c "import numpy; print numpy.get_include()" 2> /dev/null`
+  if test "x$numpy_INCLUDES" = "x"; then
+    AC_MSG_WARN([Could not find a valid numpy installation, disabling.])
+    enable_numpy=no
+  else
+    CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES -I$numpy_INCLUDES"
+    AC_CHECK_HEADER([numpy/arrayobject.h],
+      [AC_DEFINE(HAVE_NUMPY,,[whether to include numeric python support])],
+      [enable_numpy=no],
+      [#include <Python.h>])
+    if test "x$enable_numpy" != "xno"; then
+      CPPFLAGS="$save_CPPFLAGS -I$numpy_INCLUDES"
+    fi
+  fi
 fi
 
 
@@ -402,11 +411,11 @@
 $have_gtk && echo     gtk with $gtk_version API
 $have_libglade && echo     gtk.glade
 $have_gtkunixprint && echo     gtk.unixprint
-echo
 
 if test ! $have_atk || ! $have_pango || \
         ! $have_gtk || ! $have_libglade || \
         ! $have_pangocairo || ! $have_gtkunixprint; then
+  echo
   echo "The following modules will NOT be built:"
   echo
   $have_atk || echo     atk
@@ -416,3 +425,7 @@
   $have_libglade || echo     gtk.glade
   $have_gtkunixprint || echo     gtk.unixprint
 fi
+
+echo
+echo "Numpy support: $enable_numpy"
+echo
Index: setup.py
===================================================================
--- setup.py	(révision 3082)
+++ setup.py	(copie de travail)
@@ -228,16 +228,16 @@
         data_files.append((DEFS_DIR, ('pangocairo.defs',)))
         GLOBAL_MACROS.append(('HAVE_PYCAIRO',1))
 if gtk.can_build():
-    if '--disable-numeric' in sys.argv:
-        sys.argv.remove('--disable-numeric')
+    if '--disable-numpy' in sys.argv:
+        sys.argv.remove('--disable-numpy')
     else:
         try:
-            import Numeric
-            Numeric # pyflakes
+            import numpy
+            numpy # pyflakes
             GLOBAL_MACROS.append(('HAVE_NUMPY', 1))
         except ImportError:
-            print ('* Numeric module could not be found, '
-                   'will build without Numeric support.')
+            print ('* numpy module could not be found, '
+                   'will build without numpy support.')
     ext_modules.append(gtk)
     data_files.append((os.path.join(INCLUDE_DIR, 'pygtk'), ('gtk/pygtk.h',)))
     data_files.append((DEFS_DIR, ('gtk/gdk.defs', 'gtk/gdk-types.defs',
Index: README
===================================================================
--- README	(révision 3082)
+++ README	(copie de travail)
@@ -54,7 +54,7 @@
     GTK+ 2.14.0 or higher for 2.14 API
   * libglade 2.5.0 or higher (optional)
   * pycairo 1.0.2 or higher (optional)
-  * Numeric (optional)
+  * numpy (optional)
 
 This release is supporting the following GTK+ releases:
 

