--- gtk/gtk.override.orig	2009-06-25 20:52:31.000000000 -0400
+++ gtk/gtk.override	2009-06-25 20:52:53.000000000 -0400
@@ -1660,7 +1660,7 @@
     PyObject *py_use_underline = Py_True;
 
     if (!PyArg_ParseTupleAndKeywords(args, kwargs,
-                                     "|zO):GtkToggleButton.__init__", kwlist,
+                                     "|zO:GtkToggleButton.__init__", kwlist,
                                      &text, &py_use_underline))
         return -1;
     if (text)
