Introduces typo to keep a compatibility with ibus-pinyin.

To fix this typo, we should implements codes to migrate a configuration.

diff -urN ../mozc_origin/languages/pinyin/unix/ibus/mozc_engine_property.cc ./languages/pinyin/unix/ibus/mozc_engine_property.cc
--- ../mozc_origin/languages/pinyin/unix/ibus/mozc_engine_property.cc	2012-04-09 12:28:03.895173124 +0900
+++ ./languages/pinyin/unix/ibus/mozc_engine_property.cc	2012-04-09 18:44:32.013454770 +0900
@@ -73,7 +73,8 @@
     "mode.simp",
     "\xE7\xAE\x80",  // "简"
     "hiragana.png",
-    "Simplified/Traditional Chinese",
+    // TODO(hsumita): Fixes typo. s/Simplfied/Simplified/
+    "Simplfied/Traditional Chinese",
   },
 };
