diff -purN openssh-4.7p1.orig/configure.ac openssh-4.7p1/configure.ac
--- openssh-4.7p1.orig/configure.ac	2007-08-10 00:36:12.000000000 -0400
+++ openssh-4.7p1/configure.ac	2008-03-31 19:38:54.548935620 -0400
@@ -3211,6 +3211,7 @@ AC_ARG_WITH(selinux,
 		AC_CHECK_LIB(selinux, setexeccon, [ LIBSELINUX="-lselinux" ],
 		    AC_MSG_ERROR(SELinux support requires libselinux library))
 		SSHDLIBS="$SSHDLIBS $LIBSELINUX"
+		LIBS="$LIBS $LIBSELINUX"
 		AC_CHECK_FUNCS(getseuserbyname get_default_context_with_level)
 		LIBS="$save_LIBS"
 	fi ]
