diff -Nuar --exclude '*.rej' --exclude '*.orig' openssh-5.1p1+lpk/servconf.c openssh-5.1p1+lpk+glue/servconf.c
--- openssh-5.1p1+lpk/servconf.c	2008-08-23 14:37:18.000000000 -0700
+++ openssh-5.1p1+lpk+glue/servconf.c	2008-08-23 14:52:19.000000000 -0700
@@ -111,6 +111,25 @@
 	options->num_allow_groups = 0;
 	options->num_deny_groups = 0;
 	options->ciphers = NULL;
+#ifdef WITH_LDAP_PUBKEY
+ 	/* XXX dirty */
+ 	options->lpk.ld = NULL;
+ 	options->lpk.on = -1;
+ 	options->lpk.servers = NULL;
+ 	options->lpk.u_basedn = NULL;
+ 	options->lpk.g_basedn = NULL;
+ 	options->lpk.binddn = NULL;
+ 	options->lpk.bindpw = NULL;
+ 	options->lpk.sgroup = NULL;
+ 	options->lpk.filter = NULL;
+ 	options->lpk.fgroup = NULL;
+ 	options->lpk.l_conf = NULL;
+ 	options->lpk.tls = -1;
+ 	options->lpk.b_timeout.tv_sec = -1;
+ 	options->lpk.s_timeout.tv_sec = -1;
+ 	options->lpk.flags = FLAG_EMPTY;
+#endif
+
 	options->macs = NULL;
 	options->protocol = SSH_PROTO_UNKNOWN;
 	options->gateway_ports = -1;
@@ -131,25 +150,6 @@
 	options->num_permitted_opens = -1;
 	options->adm_forced_command = NULL;
 	options->chroot_directory = NULL;
-#ifdef WITH_LDAP_PUBKEY
- 	/* XXX dirty */
- 	options->lpk.ld = NULL;
- 	options->lpk.on = -1;
- 	options->lpk.servers = NULL;
- 	options->lpk.u_basedn = NULL;
- 	options->lpk.g_basedn = NULL;
- 	options->lpk.binddn = NULL;
- 	options->lpk.bindpw = NULL;
- 	options->lpk.sgroup = NULL;
- 	options->lpk.filter = NULL;
- 	options->lpk.fgroup = NULL;
- 	options->lpk.l_conf = NULL;
- 	options->lpk.tls = -1;
- 	options->lpk.b_timeout.tv_sec = -1;
- 	options->lpk.s_timeout.tv_sec = -1;
- 	options->lpk.flags = FLAG_EMPTY;
-#endif
-
 }
 
 void
