--- a/time_posix.cc.orig  2012-05-22 17:04:50.252715756 -0700
+++ b/time_posix.cc       2012-05-22 17:04:52.742723993 -0700
@@ -34,7 +34,7 @@ struct timespec TimeDelta::ToTimeSpec() 
   }
   struct timespec result =
       {seconds,
-       microseconds * Time::kNanosecondsPerMicrosecond};
+       static_cast<long int>(microseconds * Time::kNanosecondsPerMicrosecond)};
   return result;
 }
 
--- a/message_pump_libevent.cc.orig	2012-05-30 16:51:53.954509506 -0700
+++ a/message_pump_libevent.cc	2012-05-30 16:52:04.654541674 -0700
@@ -6,6 +6,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "base/auto_reset.h"
 #include "base/compiler_specific.h"
