--- tests/data/test241.orig	2008-11-20 08:12:35.000000000 +1100
+++ tests/data/test241	2009-11-05 14:22:07.000000000 +1100
@@ -33,7 +33,7 @@
 HTTP-IPv6 GET (using ip6-localhost)
  </name>
  <command>
--g "http://ip6-localhost:%HTTP6PORT/241"
+-g "http://::1:%HTTP6PORT/241"
 </command>
 <precheck>
 ./server/resolve --ipv6 ip6-localhost
@@ -48,7 +48,7 @@
 </strip>
 <protocol>
 GET /241 HTTP/1.1
-Host: ip6-localhost:%HTTP6PORT
+Host: ::1:%HTTP6PORT
 Accept: */*
 
 </protocol>
