From d14311e08510529d6dba110a47364d35bf50f94f Mon Sep 17 00:00:00 2001
From: Brandon Philips <brandon@ifup.co>
Date: Fri, 10 May 2013 19:03:15 -0700
Subject: [PATCH] disable tests

---
 tools/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tools/Makefile b/tools/Makefile
index d8c8cc0..2d41a81 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -44,7 +44,6 @@ SUBDIRS-y += remus
 SUBDIRS-$(CONFIG_X86) += xenpaging
 SUBDIRS-$(CONFIG_X86) += debugger/gdbsx
 SUBDIRS-$(CONFIG_X86) += debugger/kdd
-SUBDIRS-$(CONFIG_TESTS) += tests
 
 # These don't cross-compile
 ifeq ($(XEN_COMPILE_ARCH),$(XEN_TARGET_ARCH))
-- 
1.8.1.4

