diff -ur xen-4.1.1.orig//tools/check/check_curl xen-4.1.1/tools/check/check_curl
--- xen-4.1.1.orig//tools/check/check_curl	2011-06-15 00:03:44.000000000 +0800
+++ xen-4.1.1/tools/check/check_curl	2011-10-14 00:42:08.189717078 +0800
@@ -9,5 +9,6 @@
 fi
 
 has_or_fail curl-config
 curl_libs=`curl-config --libs` || fail "curl-config --libs failed"
+curl_libs=`echo $curl_libs | sed -re 's/-(W|march|mtune|pipe)[^[:space:]]*[[:space:]]//g'` || fail "curl-config --libs failed"
 test_link $curl_libs || fail "dependency libraries for curl are missing"
