diff -ur xen-4.2.0.orig/tools/firmware/etherboot/Makefile xen-4.2.0/tools/firmware/etherboot/Makefile
--- tools/firmware/etherboot/Makefile	2012-09-17 18:21:18.000000000 +0800
+++ tools/firmware/etherboot/Makefile	2012-11-21 14:12:54.389536642 +0800
@@ -28,13 +28,6 @@
 	$(MAKE) -C $D/src bin/$(*F).rom
 
 $T:
-	if ! wget -O _$T $(IPXE_TARBALL_URL); then \
-		$(GIT) clone $(IPXE_GIT_URL) $D.git; \
-		(cd $D.git && $(GIT) archive --format=tar --prefix=$D/ \
-		$(IPXE_GIT_TAG) | gzip >../_$T); \
-		rm -rf $D.git; \
-	fi
-	mv _$T $T
 
 $D/src/arch/i386/Makefile: $T Config
 	rm -rf $D
diff -ur xen-4.2.0.orig/tools/firmware/Makefile xen-4.2.0/tools/firmware/Makefile
--- tools/firmware/Makefile	2012-09-17 18:21:18.000000000 +0800
+++ tools/firmware/Makefile	2012-11-21 16:14:21.974868539 +0800
@@ -18,7 +18,6 @@
 	cp ovmf-makefile ovmf/Makefile;
 
 seabios-dir:
-	GIT=$(GIT) $(XEN_ROOT)/scripts/git-checkout.sh $(SEABIOS_UPSTREAM_URL) $(SEABIOS_UPSTREAM_TAG) seabios-dir
	cp seabios-config seabios-dir/.config;
 
 .PHONY: all
