--- gdata-2.0.14.orig/src/gdata/projecthosting/client.py	2011-11-17 09:22:17.350490120 -0800
+++ gdata-2.0.14/src/gdata/projecthosting/client.py	2011-11-17 09:22:44.380842001 -0800
@@ -67,8 +67,8 @@ class ProjectHostingClient(gdata.client.
       new_entry.status = gdata.projecthosting.data.Status(text=status)
 
     if owner:
-      owner = [gdata.projecthosting.data.Owner(
-          username=gdata.projecthosting.data.Username(text=owner))]
+      new_entry.owner = gdata.projecthosting.data.Owner(
+          username=gdata.projecthosting.data.Username(text=owner))
 
     if labels:
       new_entry.label = [gdata.projecthosting.data.Label(text=label)
