From 7c3b3f34f94146d5178adee4c5a184e9b1546e89 Mon Sep 17 00:00:00 2001
From: Michael Marineau <mike@marineau.org>
Date: Mon, 22 Dec 2014 21:20:46 -0800
Subject: [PATCH] disable perl scripts

---
 Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index fc76719..1fd008d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,7 +6,6 @@ NULL =
 
 SUBDIRS =		\
 	sntp		\
-	scripts		\
 	include		\
 	libntp		\
 	libparse	\
-- 
2.0.4

