Index: sbsigntool/src/sbkeysync.c
===================================================================
--- sbsigntool.orig/src/sbkeysync.c	2013-12-03 15:45:49.007312000 +0100
+++ sbsigntool/src/sbkeysync.c	2013-12-03 15:47:47.396135699 +0100
@@ -56,7 +56,8 @@
 #include "efivars.h"
 
 #define EFIVARS_MOUNTPOINT	"/sys/firmware/efi/efivars"
-#define EFIVARS_FSTYPE		0x6165676C
+#define PSTORE_FSTYPE		0x6165676C
+#define EFIVARS_FSTYPE		0xde5e81e4
 
 #define EFI_IMAGE_SECURITY_DATABASE_GUID \
 	{ 0xd719b2cb, 0x3d3a, 0x4596, \
@@ -533,7 +534,7 @@
 	if (rc)
 		return -1;
 
-	if (statbuf.f_type != EFIVARS_FSTYPE)
+	if (statbuf.f_type != EFIVARS_FSTYPE && statbuf.f_type != PSTORE_FSTYPE)
 		return -1;
 
 	return 0;
