diff --git a/Makefile.am b/Makefile.am
index d347e9b3..cc1c29b1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,2 +1,2 @@
 
-SUBDIRS = lib/ccan src docs tests
+SUBDIRS = lib/ccan src tests
diff --git a/Makefile.in b/Makefile.in
index 57e7c4a3..584d2616 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -225,7 +225,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 uuid_CFLAGS = @uuid_CFLAGS@
 uuid_LIBS = @uuid_LIBS@
-SUBDIRS = lib/ccan src docs tests
+SUBDIRS = lib/ccan src tests
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff --git a/configure b/configure
index 73198f17..94408bba 100755
--- a/configure
+++ b/configure
@@ -5517,7 +5517,7 @@ CPPFLAGS="$CPPFLAGS_save"
 EFI_CPPFLAGS=$EFI_CPPFLAGS
 
 
-ac_config_files="$ac_config_files Makefile src/Makefile lib/ccan/Makefile docs/Makefile tests/Makefile"
+ac_config_files="$ac_config_files Makefile src/Makefile lib/ccan/Makefile tests/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -6249,7 +6249,6 @@ do
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "lib/ccan/Makefile") CONFIG_FILES="$CONFIG_FILES lib/ccan/Makefile" ;;
-    "docs/Makefile") CONFIG_FILES="$CONFIG_FILES docs/Makefile" ;;
     "tests/Makefile") CONFIG_FILES="$CONFIG_FILES tests/Makefile" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
diff --git a/configure.ac b/configure.ac
index 0d8f0bbd..fb2b6c4f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -74,5 +74,5 @@ CPPFLAGS="$CPPFLAGS_save"
 AC_SUBST(EFI_CPPFLAGS, $EFI_CPPFLAGS)
 
 AC_CONFIG_FILES([Makefile src/Makefile lib/ccan/Makefile]
-		[docs/Makefile tests/Makefile])
+		[tests/Makefile])
 AC_OUTPUT
