diff -ur a/linux/Makefile b/linux/Makefile
--- a/linux/Makefile	2008-09-26 01:46:02.000000000 +0200
+++ b/linux/Makefile	2008-10-28 15:56:25.818399544 +0100
@@ -20,7 +20,7 @@
 OPTFLAGS = -g -Os
 INCLUDES = -I. -I.. -I../libinstaller
 CFLAGS	 = -W -Wall -D_FILE_OFFSET_BITS=64 $(OPTFLAGS) $(INCLUDES)
-LDFLAGS	 = -s
+LDFLAGS	 = 
 
 SRCS     = syslinux.c \
            ../libinstaller/syslxmod.c \
diff -ur a/mtools/Makefile b/mtools/Makefile
--- a/mtools/Makefile	2008-09-26 01:46:02.000000000 +0200
+++ b/mtools/Makefile	2008-10-28 15:56:45.113477480 +0100
@@ -4,7 +4,7 @@
 OPTFLAGS = -g -Os
 INCLUDES = -I. -I.. -I../libfat -I../libinstaller
 CFLAGS	 = -W -Wall -D_FILE_OFFSET_BITS=64 $(OPTFLAGS) $(INCLUDES)
-LDFLAGS	 = -s
+LDFLAGS	 = 
 
 SRCS     = syslinux.c \
 	   ../libinstaller/syslxmod.c \
diff -ur a/utils/Makefile b/utils/Makefile
--- a/utils/Makefile	2008-09-26 01:46:02.000000000 +0200
+++ b/utils/Makefile	2008-10-28 15:56:55.843968018 +0100
@@ -18,7 +18,7 @@
 include $(topdir)/MCONFIG
 
 CFLAGS   = -W -Wall -Os -fomit-frame-pointer -D_FILE_OFFSET_BITS=64
-LDFLAGS  = -O2 -s
+LDFLAGS  = -O2
 
 TARGETS	 = mkdiskimage isohybrid gethostip
 ASIS     = keytab-lilo lss16toppm md5pass ppmtolss16 sha1pass syslinux2ansi
