diff -paur xf86-video-intel-2.14.0.orig/src/intel_display.c xf86-video-intel-2.14.0.work/src/intel_display.c
--- xf86-video-intel-2.14.0.orig/src/intel_display.c	2011-01-10 15:02:16.534513000 -0800
+++ xf86-video-intel-2.14.0.work/src/intel_display.c	2011-05-19 15:06:39.732535000 -0700
@@ -57,6 +57,8 @@ struct intel_mode {
 	unsigned int fe_tv_sec;
 	unsigned int fe_tv_usec;
 
+	unsigned int last_queued_frame;
+
 	struct list outputs;
 	struct list crtcs;
 };
@@ -1435,6 +1437,12 @@ intel_do_pageflip(intel_screen_private *
 	int i, old_fb_id;
 
 	/*
+	 * Discard this flip since we already have one pending for this frame.
+	 */
+	if (  flip_info->frame <= mode->last_queued_frame )
+		return FALSE;
+
+	/*
 	 * Create a new handle for the back buffer
 	 */
 	old_fb_id = mode->fb_id;
@@ -1490,11 +1498,15 @@ intel_do_pageflip(intel_screen_private *
 	}
 
 	mode->old_fb_id = old_fb_id;
+	mode->last_queued_frame = flip_info->frame;
 	return TRUE;
 
 error_undo:
 	drmModeRmFB(mode->fd, mode->fb_id);
 	mode->fb_id = old_fb_id;
+	for (i = 0; i < config->num_crtc; i++)
+		if (config->crtc[i]->enabled)
+			mode->flip_count--;
 
 error_out:
 	xf86DrvMsg(scrn->scrnIndex, X_WARNING, "Page flip failed: %s\n",
