diff --git a/src/intel_display.c b/src/intel_display.c
index 84c7c08..a9dee28 100644
--- a/src/intel_display.c
+++ b/src/intel_display.c
@@ -400,7 +400,6 @@ intel_crtc_apply(xf86CrtcPtr crtc)
 			continue;
 
 		    intel_output = output->driver_private;
-		    intel_output_dpms_backlight(output, intel_output->dpms_mode, DPMSModeOn);
 		    intel_output->dpms_mode = DPMSModeOn;
 		}
 	}
@@ -950,9 +949,6 @@ intel_output_dpms(xf86OutputPtr output, int dpms)
 						    intel_output->output_id,
 						    props->prop_id,
 						    dpms);
-			intel_output_dpms_backlight(output,
-						      intel_output->dpms_mode,
-						      dpms);
 			intel_output->dpms_mode = dpms;
 			drmModeFreeProperty(props);
 			return;
