http://www.sqlite.org/cvstrac/tktview?tn=3951

--- test/tkt3922.test
+++ test/tkt3922.test
@@ -36,20 +36,6 @@
     }
   } {-1 integer}
 }
-do_test tkt3922.2 {
-  execsql {
-    DELETE FROM t1;
-    INSERT INTO t1 VALUES('-9223372036854775809');
-    SELECT a, typeof(a) FROM t1;
-  }
-} {-9.22337203685478e+18 real}
-do_test tkt3922.3 {
-  execsql {
-    DELETE FROM t1;
-    INSERT INTO t1 VALUES('-9223372036854776832');
-    SELECT a, typeof(a) FROM t1;
-  }
-} {-9.22337203685478e+18 real}
 do_test tkt3922.4 {
   execsql {
     DELETE FROM t1;
