diff -Naur libchrome-55729-orig/files/base/crypto/rsa_private_key_nss.cc libchrome-55729-ann/files/base/crypto/rsa_private_key_nss.cc
--- libchrome-55729-orig/files/crypto/rsa_private_key_nss.cc	2010-08-11 13:51:24.000000000 -0700
+++ libchrome-55729-ann/files/crypto/rsa_private_key_nss.cc	2010-08-11 15:46:36.000000000 -0700
@@ -10,7 +10,7 @@
 
 #include <list>
 
-#include "base/debug/leak_annotations.h"
+#define ANNOTATE_SCOPED_MEMORY_LEAK ;
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
 #include "base/string_util.h"
