From c2fd79f7dd3110001f25669e343d8b8fc835e5ce Mon Sep 17 00:00:00 2001
From: Alex Crawford <alex.crawford@coreos.com>
Date: Wed, 1 Oct 2014 10:14:41 -0700
Subject: [PATCH] move waagent.conf under /usr/share/oem

---
 waagent | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/waagent b/waagent
index 15f817d..35b3755 100644
--- a/waagent
+++ b/waagent
@@ -2547,14 +2547,14 @@ class LoadBalancerProbeServer(object):
 
 class ConfigurationProvider(object):
     """
-    Parse amd store key:values in /etc/waagent.conf.
+    Parse amd store key:values in /usr/share/oem/waagent.conf.
     """
     def __init__(self):
         self.values = dict()
-        if os.path.isfile("/etc/waagent.conf") == False:
-            raise Exception("Missing configuration in /etc/waagent.conf")
+        if os.path.isfile("/usr/share/oem/waagent.conf") == False:
+            raise Exception("Missing configuration in /usr/share/oem/waagent.conf")
         try:
-            for line in GetFileContents("/etc/waagent.conf").split('\n'):
+            for line in GetFileContents("/usr/share/oem/waagent.conf").split('\n'):
                 if not line.startswith("#") and "=" in line:
                     parts = line.split()[0].split('=')
                     value = parts[1].strip("\" ")
@@ -2563,7 +2563,7 @@ class ConfigurationProvider(object):
                     else:
                         self.values[parts[0]] = None
         except:
-            Error("Unable to parse /etc/waagent.conf")
+            Error("Unable to parse /usr/share/oem/waagent.conf")
             raise
         return
 
-- 
1.9.3

