From 8e3f94badcacfd4b681fa72619cf0c9c1d904714 Mon Sep 17 00:00:00 2001
From: Alex Crawford <alex.crawford@coreos.com>
Date: Fri, 10 Oct 2014 10:58:08 -0700
Subject: [PATCH] Allow agent to set password for 'core' user

---
 waagent | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/waagent b/waagent
index 45ab97a..6b0e0cd 100644
--- a/waagent
+++ b/waagent
@@ -972,6 +972,8 @@ class CoreOSDistro(AbstractDistro):
     CoreOS Distro concrete class
     Put CoreOS specific behavior here...
     """
+    CORE_UID = 500
+
     def __init__(self):
         super(CoreOSDistro,self).__init__()
         self.requiredDeps += [ "/usr/bin/systemctl" ]
@@ -1059,7 +1061,7 @@ class CoreOSDistro(AbstractDistro):
             pass
         if uidmin == None:
             uidmin = 100
-        if userentry != None and userentry[2] < uidmin:
+        if userentry != None and userentry[2] < uidmin and userentry[2] != self.CORE_UID:
             Error("CreateAccount: " + user + " is a system user. Will not set password.")
             return "Failed to set password for system user: " + user + " (0x06)."
         if userentry == None:
-- 
1.9.3

