Index: bus/inputcontext.c
diff --git a/bus/inputcontext.c b/bus/inputcontext.c
index 49c4a2694243f20e73929741ae25f7101b94dbc7..ec97e5499844b57d7f2cb63d039dda9d45a1a0ec 100644
--- a/bus/inputcontext.c
+++ b/bus/inputcontext.c
@@ -2015,6 +2015,9 @@ bus_input_context_unset_engine (BusInputContext *context)
             g_signal_handlers_disconnect_by_func (context->engine,
                     engine_signals[i].callback, context);
         }
+        /* focus out to let engine register properties when enabled
+           next time. */
+        bus_engine_proxy_focus_out (context->engine);
         g_object_unref (context->engine);
         context->engine = NULL;
     }
