diff -urN ibus-1.4.99.20120314.orig/src/ibusserializable.c ibus-1.4.99.20120314/src/ibusserializable.c
--- ibus-1.4.99.20120314.orig/src/ibusserializable.c	2012-11-02 16:48:00.508766644 +0900
+++ ibus-1.4.99.20120314/src/ibusserializable.c	2012-11-02 16:52:34.632899493 +0900
@@ -161,7 +161,6 @@
                                           key,
                                           attachment);
         g_variant_unref (attachment);
-        g_variant_unref (value);
     }
     g_variant_iter_free (iter);
     return 2;
