diff -ur ibus-1.4.99.20120314.orig/bus/inputcontext.c ibus-1.4.99.20120314/bus/inputcontext.c
--- ibus-1.4.99.20120314.orig/bus/inputcontext.c	2012-02-27 06:37:21.000000000 +0900
+++ ibus-1.4.99.20120314/bus/inputcontext.c	2012-04-02 15:24:41.936207367 +0900
@@ -1130,7 +1130,9 @@
     bus_input_context_clear_preedit_text (context);
     bus_input_context_update_auxiliary_text (context, text_empty, FALSE);
     bus_input_context_update_lookup_table (context, lookup_table_empty, FALSE);
-    bus_input_context_register_properties (context, props_empty);
+
+    // Workaround for http://crosbug.com/7702
+    // bus_input_context_register_properties (context, props_empty);
 
     if (context->engine) {
         bus_engine_proxy_focus_out (context->engine);
