diff -ur ibus-1.4.99.20120314.orig/bus/panelproxy.c ibus-1.4.99.20120314/bus/panelproxy.c
--- ibus-1.4.99.20120314.orig/bus/panelproxy.c	2012-02-27 06:37:21.000000000 +0900
+++ ibus-1.4.99.20120314/bus/panelproxy.c	2012-04-02 15:36:12.633725645 +0900
@@ -481,7 +481,7 @@
 
     g_return_if_fail (panel->focused_context == context);
 
-    bus_panel_proxy_set_cursor_location (panel, x, y, w, h);
+    // bus_panel_proxy_set_cursor_location (panel, x, y, w, h);
 }
 
 static void
Only in ibus-1.4.99.20120314/bus: panelproxy.c.orig
