diff -rupN nss-3.12.8/mozilla/security/nss/lib/certdb/certdb.c nss-3.12.8_patched/mozilla/security/nss/lib/certdb/certdb.c
--- nss-3.12.8/mozilla/security/nss/lib/certdb/certdb.c	2010-09-02 00:52:02.000000000 +0000
+++ nss-3.12.8_patched/mozilla/security/nss/lib/certdb/certdb.c	2012-04-02 20:58:31.821621891 +0000
@@ -3030,6 +3030,7 @@ cert_InitLocks(void)
         PORT_Assert(certTrustLock != NULL);
         if (!certTrustLock) {
             PZ_DestroyLock(certRefCountLock);
+            certRefCountLock = NULL;
             return SECFailure;
         }
     }    
