Fix syscalls for arm64

diff --git a/display.c b/display.c
index a5cb2c2..98278b2 100644
--- a/display.c
+++ b/display.c
@@ -61,17 +61,14 @@ Display_call_s Display_call[] = {
 	{ "pwrite:", __NR_pwrite64 },
 	{ "sync:  ", __NR_sync },
 	{ "fsync: ", __NR_fsync },
-	{ "open:  ", __NR_open },
 	{ "close: ", __NR_close },
+	{ "fstat: ", __NR_fstat },
+#if defined(__x86_64__)
+	{ "open:  ", __NR_open },
 	{ "creat: ", __NR_creat },
 	{ "unlink:", __NR_unlink },
-#ifdef __x86_64__
 	{ "stat:  ", __NR_stat },
-	{ "fstat: ", __NR_fstat },
 	{ "lstat: ", __NR_lstat },
-#else
-	{ "stat:  ", __NR_stat64 },
-	{ "fstat: ", __NR_fstat64 },
 #endif
 	{ NULL, 0 }};
 
