diff -uNr alsa-utils-1.0.23.ORIG//alsaconf/alsaconf.in alsa-utils-1.0.23/alsaconf/alsaconf.in
--- alsa-utils-1.0.23.ORIG//alsaconf/alsaconf.in	2010-04-16 23:16:24.972269218 +0100
+++ alsa-utils-1.0.23/alsaconf/alsaconf.in	2010-04-16 23:16:47.781310369 +0100
@@ -301,7 +301,7 @@
   fi
 else
 if [ "$distribution" = "gentoo" ]; then
-  cfgfile="/etc/modules.d/alsa"
+  cfgfile="/etc/modprobe.d/alsa.conf"
 elif [ "$kernel" = "new" ]; then
   cfgfile="/etc/modprobe.conf"
   if [ -d /etc/modprobe.d ]; then
