? cups/.deps
? cups/Makefile
? cups/Makefile.in
Index: cups/Makefile.am
===================================================================
RCS file: /cvsroot/bluez/utils/cups/Makefile.am,v
retrieving revision 1.9
diff -u -r1.9 Makefile.am
--- cups/Makefile.am	20 Aug 2006 02:21:03 -0000	1.9
+++ cups/Makefile.am	1 Jun 2007 15:47:14 -0000
@@ -1,6 +1,6 @@
 
 if CUPS
-cupsdir = $(libdir)/cups/backend
+cupsdir = `cups-config --serverbin`/backend
 
 cups_PROGRAMS = bluetooth
 else
