diff -ur pambase-20120417.orig/system-auth.in pambase-20120417/system-auth.in
--- pambase-20120417.orig/system-auth.in	2016-05-15 21:01:29.705547981 -0700
+++ pambase-20120417/system-auth.in	2016-05-15 21:03:08.134220108 -0700
@@ -7,14 +7,16 @@
 #if HAVE_KRB5
 auth		KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
 #endif
-auth		required	pam_unix.so try_first_pass LIKEAUTH nullok DEBUG
+auth		sufficient	pam_sss.so use_first_pass
+auth		sufficient	pam_unix.so try_first_pass LIKEAUTH nullok DEBUG
 /* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
-auth		optional	pam_permit.so
+auth		required	pam_deny.so
  
 #if HAVE_KRB5
 account		KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
 #endif
 account		required	pam_unix.so DEBUG
+account		sufficient	pam_sss.so
 /* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
 account		optional	pam_permit.so
  
@@ -27,9 +29,10 @@
 #if HAVE_KRB5
 password	KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
 #endif
-password	required	pam_unix.so try_first_pass UNIX_AUTHTOK nullok UNIX_EXTENDED_ENCRYPTION DEBUG
+password	sufficient	pam_unix.so try_first_pass UNIX_AUTHTOK nullok UNIX_EXTENDED_ENCRYPTION DEBUG
+password	sufficient	pam_sss.so use_authtok
 /* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
-password	optional	pam_permit.so
+password	required	pam_deny.so
  
 #if HAVE_PAM_SSH
 session		optional	pam_ssh.so
diff -ur pambase-20120417.orig/system-session.inc pambase-20120417/system-session.inc
--- pambase-20120417.orig/system-session.inc	2012-03-20 11:10:47.000000000 -0700
+++ pambase-20120417/system-session.inc	2016-05-15 21:02:00.151753215 -0700
@@ -23,3 +23,4 @@
 #endif
 
 session		optional	pam_permit.so
+session		optional	pam_sss.so
