diff -ruN laptop-mode-tools-1.59/etc/laptop-mode/conf.d/ac97-powersave.conf laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/ac97-powersave.conf
--- laptop-mode-tools-1.59/etc/laptop-mode/conf.d/ac97-powersave.conf
+++ laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/ac97-powersave.conf
@@ -18,7 +18,7 @@
 ###############################################################################
 
 # Control AC97 audio chipset power?
-CONTROL_AC97_POWER="auto"
+CONTROL_AC97_POWER=1
 
 # Enable debug mode for this module
 # Set to 1 if you want to debug this module
diff -ruN laptop-mode-tools-1.59/etc/laptop-mode/conf.d/intel-hda-powersave.conf laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/intel-hda-powersave.conf
--- laptop-mode-tools-1.59/etc/laptop-mode/conf.d/intel-hda-powersave.conf
+++ laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/intel-hda-powersave.conf
@@ -20,7 +20,7 @@
 
 # Control INTEL HDA audio chipset power?
 # Set to 0 to disable
-CONTROL_INTEL_HDA_POWER="auto"
+CONTROL_INTEL_HDA_POWER=1
 
 # Handle power savings for Intel HDA under specific circumstances
 BATT_INTEL_HDA_POWERSAVE=1
diff -ruN laptop-mode-tools-1.59/etc/laptop-mode/conf.d/intel-sata-powermgmt.conf laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/intel-sata-powermgmt.conf
--- laptop-mode-tools-1.59/etc/laptop-mode/conf.d/intel-sata-powermgmt.conf
+++ laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/intel-sata-powermgmt.conf
@@ -20,7 +20,7 @@
 
 # Control Intel SATA chipset power management?
 # Set to 0 to disable
-CONTROL_INTEL_SATA_POWER="auto"
+CONTROL_INTEL_SATA_POWER=1
 
 # Handle power management of the Intel SATA deivce under specific circumstances
 BATT_ACTIVATE_SATA_POWER=1
diff -ruN laptop-mode-tools-1.59/etc/laptop-mode/conf.d/usb-autosuspend.conf laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/usb-autosuspend.conf
--- laptop-mode-tools-1.59/etc/laptop-mode/conf.d/usb-autosuspend.conf
+++ laptop-mode-tools-1.59.new/etc/laptop-mode/conf.d/usb-autosuspend.conf
@@ -24,7 +24,7 @@
 
 # Enable USB autosuspend feature?
 # Set to 0 to disable
-CONTROL_USB_AUTOSUSPEND="auto"
+CONTROL_USB_AUTOSUSPEND=1
 
 # The list of USB IDs that should not use autosuspend. Use lsusb to find out the
 # IDs of your USB devices.
diff -ruN laptop-mode-tools-1.59/etc/laptop-mode/laptop-mode.conf laptop-mode-tools-1.59.new/etc/laptop-mode/laptop-mode.conf
--- laptop-mode-tools-1.59/etc/laptop-mode/laptop-mode.conf
+++ laptop-mode-tools-1.59.new/etc/laptop-mode/laptop-mode.conf
@@ -283,7 +283,7 @@
 # Should laptop mode tools control the hard drive power management settings?
 #
 # Set to 0 to disable
-CONTROL_HD_POWERMGMT="auto"
+CONTROL_HD_POWERMGMT=1
 
 
 #
