--- laptop-mode-tools-1.59/usr/sbin/laptop_mode
+++ laptop-mode-tools-1.59/usr/sbin/laptop_mode
@@ -165,7 +165,7 @@
         elif [ "$1" = "ERR" ]; then
             logger -p $SYSLOG_FACILITY.err -t laptop-mode "$2";
         elif [ "$1" = "VERBOSE" ]; then
-            if [ x$VERBOSE_OUTPUT = x1 ]; then
+            if [ x$VERBOSE_OUTPUT = x1 -a "$DEBUG" = 1 ]; then
                 logger -p $SYSLOG_FACILITY.debug -t laptop-mode "$2";
             fi
         else
