--- laptop-mode-tools-1.59/etc/laptop-mode/conf.d/usb-autosuspend.conf
+++ laptop-mode-tools-1.59/etc/laptop-mode/conf.d/usb-autosuspend.conf
@@ -34,7 +34,7 @@
 # The list of USB driver types that should not use autosuspend.  The driver
 # type is given by "DRIVER=..." in a USB device's uevent file.
 # Example: AUTOSUSPEND_USBID_BLACKLIST="usbhid usb-storage"
-AUTOSUSPEND_USBTYPE_BLACKLIST=""
+AUTOSUSPEND_USBTYPE_BLACKLIST="usbhid"
 
 # Trigger auto-suspension of the USB deivce under conditional circumstances
 BATT_SUSPEND_USB=1
