diff -c -r mozilla/security/nss/lib.orig/nss/nss.def mozilla/security/nss/lib/nss/nss.def
*** mozilla/security/nss/lib.orig/nss/nss.def	2010-04-30 00:47:47.000000000 -0700
--- mozilla/security/nss/lib/nss/nss.def	2011-11-29 16:33:32.828968066 -0800
***************
*** 647,652 ****
--- 647,653 ----
  PK11_GetPQGParamsFromPrivateKey;
  PK11_GetPrivateKeyNickname;
  PK11_GetPublicKeyNickname;
+ PK11_GetCertificateNickname;
  PK11_GetSymKeyNickname;
  PK11_ImportDERPrivateKeyInfoAndReturnKey;
  PK11_ImportPrivateKeyInfoAndReturnKey;
***************
*** 658,663 ****
--- 659,665 ----
  PK11_ProtectedAuthenticationPath;
  PK11_SetPrivateKeyNickname;
  PK11_SetPublicKeyNickname;
+ PK11_SetCertificateNickname;
  PK11_SetSymKeyNickname;
  SECKEY_DecodeDERSubjectPublicKeyInfo;
  SECKEY_DestroyPublicKeyList;
diff -r -c mozilla/security/nss/lib.orig/pk11wrap/pk11akey.c mozilla/security/nss/lib/pk11wrap/pk11akey.c
*** mozilla/security/nss/lib.orig/pk11wrap/pk11akey.c	2011-11-29 12:16:56.000000000 -0800
--- mozilla/security/nss/lib/pk11wrap/pk11akey.c	2011-11-29 13:16:22.672540042 -0800
***************
*** 1907,1912 ****
--- 1907,1919 ----
      return PK11_GetObjectNickname(pubKey->pkcs11Slot,pubKey->pkcs11ID);
  }
  
+ char *
+ PK11_GetCertificateNickname(CERTCertificate *certificate)
+ {
+     return PK11_GetObjectNickname(certificate->slot,
+ 					certificate->pkcs11ID);
+ }
+ 
  SECStatus
  PK11_SetPrivateKeyNickname(SECKEYPrivateKey *privKey, const char *nickname)
  {
***************
*** 1921,1926 ****
--- 1928,1940 ----
  					pubKey->pkcs11ID,nickname);
  }
  
+ SECStatus
+ PK11_SetCertificateNickname(CERTCertificate *certificate, const char *nickname)
+ {
+     return PK11_SetObjectNickname(certificate->slot,
+ 					certificate->pkcs11ID,nickname);
+ }
+ 
  SECKEYPQGParams *
  PK11_GetPQGParamsFromPrivateKey(SECKEYPrivateKey *privKey)
  {
diff -r -c mozilla/security/nss/lib.orig/pk11wrap/pk11obj.c mozilla/security/nss/lib/pk11wrap/pk11obj.c
*** mozilla/security/nss/lib.orig/pk11wrap/pk11obj.c	2009-02-15 19:47:21.000000000 -0800
--- mozilla/security/nss/lib/pk11wrap/pk11obj.c	2011-11-29 11:54:01.946485857 -0800
***************
*** 1396,1402 ****
  	slot = ((PK11SymKey *)objSpec)->slot;
  	handle = ((PK11SymKey *)objSpec)->objectID;
  	break;
!     case PK11_TypeCert: /* don't handle cert case for now */
      default:
  	break;
      }
--- 1396,1405 ----
  	slot = ((PK11SymKey *)objSpec)->slot;
  	handle = ((PK11SymKey *)objSpec)->objectID;
  	break;
!     case PK11_TypeCert:
! 	slot = ((CERTCertificate *)objSpec)->slot;
! 	handle = ((CERTCertificate *)objSpec)->pkcs11ID;
! 	break;
      default:
  	break;
      }
***************
*** 1446,1452 ****
  	slot = ((PK11SymKey *)objSpec)->slot;
  	handle = ((PK11SymKey *)objSpec)->objectID;
  	break;
!     case PK11_TypeCert: /* don't handle cert case for now */
      default:
  	break;
      }
--- 1449,1458 ----
  	slot = ((PK11SymKey *)objSpec)->slot;
  	handle = ((PK11SymKey *)objSpec)->objectID;
  	break;
!     case PK11_TypeCert:
! 	slot = ((CERTCertificate *)objSpec)->slot;
! 	handle = ((CERTCertificate *)objSpec)->pkcs11ID;
! 	break;
      default:
  	break;
      }
diff -r -c mozilla/security/nss/lib.orig/pk11wrap/pk11pub.h mozilla/security/nss/lib/pk11wrap/pk11pub.h
*** mozilla/security/nss/lib.orig/pk11wrap/pk11pub.h	2010-04-25 16:37:39.000000000 -0700
--- mozilla/security/nss/lib/pk11wrap/pk11pub.h	2011-11-29 11:53:04.605709449 -0800
***************
*** 462,472 ****
--- 462,475 ----
  char * PK11_GetSymKeyNickname(PK11SymKey *symKey);
  char * PK11_GetPrivateKeyNickname(SECKEYPrivateKey *privKey);
  char * PK11_GetPublicKeyNickname(SECKEYPublicKey *pubKey);
+ char * PK11_GetCertificateNickname(CERTCertificate *certificate);
  SECStatus PK11_SetSymKeyNickname(PK11SymKey *symKey, const char *nickname);
  SECStatus PK11_SetPrivateKeyNickname(SECKEYPrivateKey *privKey, 
  							const char *nickname);
  SECStatus PK11_SetPublicKeyNickname(SECKEYPublicKey *pubKey, 
  							const char *nickname);
+ SECStatus PK11_SetCertificateNickname(CERTCertificate *certificate,
+ 							const char *nickname);
  
  /* size to hold key in bytes */
  unsigned int PK11_GetKeyLength(PK11SymKey *key);
