diff -rupN nss-3.12.8/mozilla/security/nss/lib/freebl/unix_rand.c nss-3.12.8_patched/mozilla/security/nss/lib/freebl/unix_rand.c
--- nss-3.12.8/mozilla/security/nss/lib/freebl/unix_rand.c	2010-04-29 17:20:02.000000000 -0700
+++ nss-3.12.8_patched/mozilla/security/nss/lib/freebl/unix_rand.c	2012-04-18 10:38:53.013840985 -0700
@@ -949,6 +949,17 @@ void RNG_SystemInfoForRNG(void)
     || defined(HPUX)
     if (bytes)
         return;
+
+    /*
+     * Modified to abort the process on Chromium OS if it failed
+     * to read from /dev/urandom.
+     *
+     * See crosbug.com/29623 for details.
+     */
+    fprintf(stderr, "[ERROR:%s(%d)] NSS failed to read from /dev/urandom. "
+            "Abort process.\n", __FILE__, __LINE__);
+    fflush(stderr);
+    abort();
 #endif
 
 #ifdef SOLARIS
