diff -aurN nss-3.14-urandom/mozilla/security/nss/lib/freebl/unix_rand.c nss-3.14/mozilla/security/nss/lib/freebl/unix_rand.c
--- nss-3.14-urandom/mozilla/security/nss/lib/freebl/unix_rand.c	2012-12-28 16:31:12.017070243 -0800
+++ nss-3.14/mozilla/security/nss/lib/freebl/unix_rand.c	2012-12-28 16:31:49.107466816 -0800
@@ -925,6 +925,17 @@
     || defined(HPUX)
     if (bytes)
         return;
+
+    /*
+     * Modified to abort the process on Chromium OS if it failed
+     * to read from /dev/urandom.
+     *
+     * See crosbug.com/29623 for details.
+     */
+    fprintf(stderr, "[ERROR:%s(%d)] NSS failed to read from /dev/urandom. "
+            "Abort process.\n", __FILE__, __LINE__);
+    fflush(stderr);
+    abort();
 #endif
 
 #ifdef SOLARIS
