*** mozilla/security/nss/lib/libpkix/pkix_pl_nss/module/pkix_pl_pk11certstore.c	2012-08-23 14:37:23.300678262 -0700
--- mozilla/security/nss/lib/libpkix/pkix_pl_nss/module/pkix_pl_pk11certstore.c.new	2012-08-23 17:52:26.209075001 -0700
***************
*** 275,292 ****
                      !(CERT_LIST_END(node, pk11CertList));
                      node = CERT_LIST_NEXT(node)) {
  
!                         PKIX_PL_NSSCALLRV
!                                 (CERTSTORE,
!                                 nssCert,
!                                 CERT_NewTempCertificate,
!                                         (dbHandle,
!                                         &(node->cert->derCert),
!                                         NULL, /* nickname */
!                                         PR_FALSE,
!                                         PR_TRUE)); /* copyDER */
! 
                          if (!nssCert) {
!                                 continue; /* just skip bad certs */
                          }
  
                          PKIX_CHECK_ONLY_FATAL(pkix_pl_Cert_CreateWithNSSCert
--- 275,283 ----
                      !(CERT_LIST_END(node, pk11CertList));
                      node = CERT_LIST_NEXT(node)) {
  
!                         nssCert = CERT_DupCertificate(node->cert);
                          if (!nssCert) {
!                                 continue;
                          }
  
                          PKIX_CHECK_ONLY_FATAL(pkix_pl_Cert_CreateWithNSSCert
