From e07c6f2ea4963e6f9cdf6ee5eceaaf6f30319eda Mon Sep 17 00:00:00 2001
From: David Michael <david.michael@coreos.com>
Date: Tue, 17 Jan 2017 13:20:43 -0800
Subject: [PATCH] gentoolkit/glsa: Fix AttributeError from int coercion

---
 pym/gentoolkit/glsa/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pym/gentoolkit/glsa/__init__.py b/pym/gentoolkit/glsa/__init__.py
index ba1eed7..0ed539e 100644
--- a/pym/gentoolkit/glsa/__init__.py
+++ b/pym/gentoolkit/glsa/__init__.py
@@ -567,6 +567,8 @@ class Glsa:
 				(self.revised, count) = self.revised.split(":")
 			else:
 				count = 1
+		else:
+			count = count.value
 
 		self.revised = format_date(self.revised)
 
-- 
2.7.4

