diff --git a/htpdate.c b/htpdate.c
index 7524ce2..aa66e9f 100644
--- a/htpdate.c
+++ b/htpdate.c
@@ -67,7 +67,7 @@
 #define	URLSIZE					128
 #define	BUFFERSIZE				1024
 
-#define ERR_TIMESTAMP           0x7FFFFFFF      /* Err fetching date in getHTTPdate */
+#define ERR_TIMESTAMP           LONG_MAX        /* Err fetching date in getHTTPdate */
 
 #define sign(x) (x < 0 ? (-1) : 1)
 
@@ -82,8 +82,8 @@ static int              waitfornw = 0;
 static int              recvdvalidresp = 0;
 
 /* Insertion sort is more efficient (and smaller) than qsort for small lists */
-static void insertsort( int a[], int length ) {
-	int i, j, value;
+static void insertsort( long a[], int length ) {
+	long i, j, value;
 
 	for ( i = 1; i < length; i++ ) {
 		value = a[i];
@@ -535,14 +535,14 @@ int main( int argc, char *argv[] ) {
 	char				*user = NULL, *userstr = NULL, *group = NULL;
 	long long			sumtimes;
 	double				timeavg, drift = 0;
-	int					timedelta[MAX_HTTP_HOSTS], timestamp;
-	int                 numservers, validtimes, goodtimes, mean;
+	long				timedelta[MAX_HTTP_HOSTS], timestamp;
+	long				numservers, validtimes, goodtimes, mean;
 	int					nap = 0, when = 500000, precision = 0;
 	int					setmode = 0, burstmode = 0, try, offsetdetect;
 	int					i, burst, param;
 	int					daemonize = 0;
 	int					ipversion = DEFAULT_IP_VERSION;
-	int					timelimit = DEFAULT_TIME_LIMIT;
+	long				timelimit = DEFAULT_TIME_LIMIT;
 	int					minsleep = DEFAULT_MIN_SLEEP;
 	int					maxsleep = DEFAULT_MAX_SLEEP;
 	int					sleeptime = minsleep;
@@ -608,7 +608,7 @@ int main( int argc, char *argv[] ) {
 			setmode = 2;
 			break;
 		case 't':			/* disable "sanity" time check */
-			timelimit = 2100000000;
+			timelimit = LONG_MAX - 1;
 			break;
 		case 'u':			/* drop root privileges and run as user */
 			user = (char *)optarg;
