diff --git a/htpdate.c b/htpdate.c
index d0e7fbb..adf351f 100644
--- a/htpdate.c
+++ b/htpdate.c
@@ -65,7 +65,7 @@
 #define	MAX_ATTEMPT				2				/* Poll attempts */
 #define	DEFAULT_PID_FILE		"/var/run/htpdate.pid"
 #define	URLSIZE					128
-#define	BUFFERSIZE				1024
+#define	BUFFERSIZE				8192
 
 #define ERR_TIMESTAMP           LONG_MAX        /* Err fetching date in getHTTPdate */
 
@@ -246,7 +246,7 @@ static long getHTTPdate( char *host, char *port, char *proxy, char *proxyport, c
 	/* Receive data from the web server
 	   The return code from recv() is the number of bytes received
 	*/
-	buffer_size = recv(server_s, buffer, BUFFERSIZE - 1, 0);
+	buffer_size = recv(server_s, buffer, BUFFERSIZE - 1, MSG_WAITALL);
 	if ( buffer_size >= 0 ) {
 		/* NUL terminate the string data. */
 		buffer[buffer_size] = '\0';
