From 2c3963923f465ae7246b819a1770dac20be48a04 Mon Sep 17 00:00:00 2001
From: Benjamin Gilbert <benjamin.gilbert@coreos.com>
Date: Wed, 3 Jan 2018 17:01:55 -0800
Subject: [PATCH 6/6] arch/x86/kernel: try stubbing out free_ldt_pgtables

---
 arch/x86/kernel/ldt.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/arch/x86/kernel/ldt.c b/arch/x86/kernel/ldt.c
index 26d713ecad34..38e5429b0781 100644
--- a/arch/x86/kernel/ldt.c
+++ b/arch/x86/kernel/ldt.c
@@ -199,18 +199,6 @@ map_ldt_struct(struct mm_struct *mm, struct ldt_struct *ldt, int slot)
 
 static void free_ldt_pgtables(struct mm_struct *mm)
 {
-#ifdef CONFIG_PAGE_TABLE_ISOLATION
-	struct mmu_gather tlb;
-	unsigned long start = LDT_BASE_ADDR;
-	unsigned long end = start + (1UL << PGDIR_SHIFT);
-
-	if (!static_cpu_has(X86_FEATURE_PTI))
-		return;
-
-	tlb_gather_mmu(&tlb, mm, start, end);
-	free_pgd_range(&tlb, start, end, start, end);
-	tlb_finish_mmu(&tlb, start, end);
-#endif
 }
 
 /* After calling this, the LDT is immutable. */
-- 
2.13.6

