--- talloc-2.0.0-orig/configure.ac	2009-08-31 02:58:36 +0900
+++ talloc-2.0.0/configure.ac	2009-10-23 11:00:13 +0900
@@ -36,10 +36,15 @@
 m4_include(libtalloc.m4)
 m4_include(compat/talloc_compat1.m4)
 
-AC_PATH_PROG(XSLTPROC,xsltproc)
 DOC_TARGET=""
-if test -n "$XSLTPROC"; then
-	DOC_TARGET=doc
+AC_ARG_WITH([doc],
+	AS_HELP_STRING([--without-doc], [disable manpage generation]),
+	[], [with_doc=yes])
+if test "yes" = "$with_doc"; then
+	AC_PATH_PROG(XSLTPROC,xsltproc)
+	if test -n "$XSLTPROC"; then
+		DOC_TARGET=doc
+	fi
 fi
 AC_SUBST(DOC_TARGET)
 
