diff -ruN OpenCV-2.3.0-1-ol1/CMakeLists.txt OpenCV-2.3.0-2-numpy/CMakeLists.txt
--- OpenCV-2.3.0-1-ol1/CMakeLists.txt	2011-08-28 14:53:46.000000000 +0200
+++ OpenCV-2.3.0-2-numpy/CMakeLists.txt	2011-08-28 15:26:37.000000000 +0200
@@ -640,7 +640,7 @@
 
 if(PYTHON_NUMPY_PROCESS EQUAL 0)
     set(PYTHON_USE_NUMPY 1)
-    add_definitions(-D PYTHON_USE_NUMPY=1)
+    add_definitions(-DPYTHON_USE_NUMPY=1)
     include_directories(AFTER ${PYTHON_NUMPY_INCLUDE_DIRS})
     message(STATUS "    Use NumPy headers from: ${PYTHON_NUMPY_INCLUDE_DIRS}")
 else()
