diff --git a/src/main.conf b/src/main.conf
index 321f622..f6784fb 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -38,7 +38,7 @@ AutoConnectTimeout = 60
 
 # What value should be assumed for the adapter Powered property when
 # SetProperty(Powered, ...) hasn't been called yet. Defaults to true
-InitiallyPowered = true
+InitiallyPowered = false
 
 # Remember the previously stored Powered state when initializing adapters
 RememberPowered = true
