diff --git a/src/bluetooth.conf b/src/bluetooth.conf
index 664dbd9..3263112 100644
--- a/src/bluetooth.conf
+++ b/src/bluetooth.conf
@@ -7,7 +7,7 @@
 
   <!-- ../system.conf have denied everything, so we just punch some holes -->
 
-  <policy user="root">
+  <policy user="bluetooth">
     <allow own="org.bluez"/>
     <allow send_destination="org.bluez"/>
     <allow send_interface="org.bluez.Agent"/>
@@ -18,13 +18,7 @@
     <allow send_interface="org.bluez.ThermometerWatcher"/>
   </policy>
 
-  <policy at_console="true">
-    <allow send_destination="org.bluez"/>
-  </policy>
-
-  <!-- allow users of lp group (printing subsystem) to 
-       communicate with bluetoothd -->
-  <policy group="lp">
+  <policy user="chronos">
     <allow send_destination="org.bluez"/>
   </policy>
 
