diff -ur setools-3.3.8.orig/configure.ac setools-3.3.8/configure.ac
--- setools-3.3.8.orig/configure.ac	2013-01-16 08:36:24.000000000 -0800
+++ setools-3.3.8/configure.ac	2015-06-11 15:01:16.476072420 -0700
@@ -593,25 +593,6 @@
 	        sepol_new_errcodes="yes",
                 sepol_new_errcodes="no")
 
-AC_RUN_IFELSE(
-   [AC_LANG_SOURCE([
-#include <sepol/policydb/policydb.h>
-#include <stdio.h>
-#include <stdlib.h>
-int main(void) {
-    FILE *f = fopen("conftest.data", "w");
-    if (f != NULL && fprintf(f, "%d", POLICYDB_VERSION_MAX) > 0) {
-       fclose(f);
-       exit(EXIT_SUCCESS);
-    }
-    exit(EXIT_FAILURE);
-}])],
-    sepol_policy_version_max=`cat conftest.data`,
-    AC_MSG_FAILURE([could not determine maximum libsepol policy version]))
-AC_DEFINE_UNQUOTED(SEPOL_POLICY_VERSION_MAX, ${sepol_policy_version_max}, [maximum policy version supported by libsepol])
-CFLAGS="${sepol_save_CFLAGS}"
-CPPFLAGS="${sepol_save_CPPFLAGS}"
-
 if test ${use_selinux} = "yes"; then
 dnl Locate selinux policy root directory
     AC_MSG_CHECKING([for selinux policy root])
