diff -Naur nspr-4.8.6_orig/mozilla/nsprpub/configure nspr-4.8.6/mozilla/nsprpub/configure
--- nspr-4.8.6_orig/mozilla/nsprpub/configure	2011-07-06 16:06:57.000000000 -0700
+++ nspr-4.8.6/mozilla/nsprpub/configure	2011-07-06 16:15:57.000000000 -0700
@@ -1241,16 +1241,16 @@
 fi
 
 if test -z "$SKIP_COMPILER_CHECKS"; then
-if test "$target" != "$host"; then
-    echo "cross compiling from $host to $target"
+if test "$target" != "$build"; then
+    echo "cross compiling from $build to $target"
     cross_compiling=yes
 
     _SAVE_CC="$CC"
     _SAVE_CFLAGS="$CFLAGS"
     _SAVE_LDFLAGS="$LDFLAGS"
 
-    echo $ac_n "checking for $host compiler""... $ac_c" 1>&6
-echo "configure:1254: checking for $host compiler" >&5
+    echo $ac_n "checking for $build compiler""... $ac_c" 1>&6
+echo "configure:1254: checking for $build compiler" >&5
     for ac_prog in $HOST_CC gcc cc /usr/ucb/cc
 do
 # Extract the first word of "$ac_prog", so it can be a program name with args.
@@ -1301,8 +1301,8 @@
     CFLAGS="$HOST_CFLAGS"
     LDFLAGS="$HOST_LDFLAGS"
 
-    echo $ac_n "checking whether the $host compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works""... $ac_c" 1>&6
-echo "configure:1306: checking whether the $host compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works" >&5
+    echo $ac_n "checking whether the $build compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works""... $ac_c" 1>&6
+echo "configure:1306: checking whether the $build compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works" >&5
     cat > conftest.$ac_ext <<EOF
 #line 1308 "configure"
 #include "confdefs.h"
@@ -1318,7 +1318,7 @@
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
   rm -rf conftest*
-  { echo "configure: error: installation or configuration problem: $host compiler $HOST_CC cannot create executables." 1>&2; exit 1; } 
+  { echo "configure: error: installation or configuration problem: $build compiler $HOST_CC cannot create executables." 1>&2; exit 1; } 
 fi
 rm -f conftest*
 
@@ -2703,6 +2703,9 @@
     i?86-apple-darwin*:powerpc-apple-darwin*)
                                 cross_compiling=yes
         ;;
+    x86_64-*:i686-*)
+                                cross_compiling=yes
+        ;;
 esac
 
 if test "$cross_compiling"  = "yes"; then
