diff --git a/common.mk b/common.mk
index 357d1f9..9a91c43 100644
--- a/common.mk
+++ b/common.mk
@@ -260,7 +260,7 @@ endif
 #  CXXFLAGS := -mahflag $(CXXFLAGS) # Prepend to the list
 #  CXXFLAGS := $(filter-out badflag,$(CXXFLAGS)) # Filter out a value
 # The same goes for CFLAGS.
-COMMON_CFLAGS := -Wall -Werror -fstack-protector-strong -fno-strict-aliasing \
+COMMON_CFLAGS := -Wall -Werror -fstack-protector-all -fno-strict-aliasing \
   -ggdb3 -Wa,--noexecstack -O1 -fvisibility=internal -Wformat=2
 CXXFLAGS += $(COMMON_CFLAGS)
 CFLAGS += $(COMMON_CFLAGS)
