--- Makefile.in
+++ Makefile.in
@@ -47,6 +47,7 @@
 SVN_SASL_LIBS = @SVN_SASL_LIBS@
 SVN_SERF_LIBS = @SVN_SERF_LIBS@
 SVN_SQLITE_LIBS = @SVN_SQLITE_LIBS@
+SVN_XML_LIBS = -lexpat
 SVN_ZLIB_LIBS = @SVN_ZLIB_LIBS@
 
 LIBS = @LIBS@
--- build/ac-macros/aprutil.m4
+++ build/ac-macros/aprutil.m4
@@ -77,16 +77,14 @@
     AC_MSG_ERROR([apu-config --prefix failed])
   fi
 
-  dnl When APR stores the dependent libs in the .la file, we don't need
-  dnl --libs.
-  SVN_APRUTIL_LIBS="`$apu_config --link-libtool --libs`"
+  SVN_APRUTIL_LIBS="`$apu_config --link-libtool`"
   if test $? -ne 0; then
-    AC_MSG_ERROR([apu-config --link-libtool --libs failed])
+    AC_MSG_ERROR([apu-config --link-libtool failed])
   fi
 
-  SVN_APRUTIL_EXPORT_LIBS="`$apu_config --link-ld --libs`"
+  SVN_APRUTIL_EXPORT_LIBS="`$apu_config --link-ld`"
   if test $? -ne 0; then
-    AC_MSG_ERROR([apu-config --link-ld --libs failed])
+    AC_MSG_ERROR([apu-config --link-ld failed])
   fi
 
   AC_SUBST(SVN_APRUTIL_INCLUDES)
--- build/ac-macros/apr.m4
+++ build/ac-macros/apr.m4
@@ -74,16 +74,14 @@
     AC_MSG_ERROR([apr-config --prefix failed])
   fi
 
-  dnl When APR stores the dependent libs in the .la file, we don't need 
-  dnl --libs.
-  SVN_APR_LIBS="`$apr_config --link-libtool --libs`"
+  SVN_APR_LIBS="`$apr_config --link-libtool`"
   if test $? -ne 0; then
-    AC_MSG_ERROR([apr-config --link-libtool --libs failed])
+    AC_MSG_ERROR([apr-config --link-libtool failed])
   fi
 
-  SVN_APR_EXPORT_LIBS="`$apr_config --link-ld --libs`"
+  SVN_APR_EXPORT_LIBS="`$apr_config --link-ld`"
   if test $? -ne 0; then
-    AC_MSG_ERROR([apr-config --link-ld --libs failed])
+    AC_MSG_ERROR([apr-config --link-ld failed])
   fi
 
   SVN_APR_SHLIB_PATH_VAR="`$apr_config --shlib-path-var`"
