diff -ru systemd-209/Makefile.am systemd-209-modified/Makefile.am
--- systemd-209/Makefile.am	2014-02-20 00:55:28.982451923 +0100
+++ systemd-209-modified/Makefile.am	2014-02-20 19:23:53.715184928 +0100
@@ -4429,13 +4429,6 @@
 	libsystemd-login.la \
 	libsystemd-id128.la \
 	libsystemd-daemon.la
-
-pkgconfiglib_DATA += \
-	src/compat-libs/libsystemd-journal.pc \
-	src/compat-libs/libsystemd-login.pc \
-	src/compat-libs/libsystemd-id128.pc \
-	src/compat-libs/libsystemd-daemon.pc
-
 # move lib from $(libdir) to $(rootlibdir) and update devel link, if needed
 compat-lib-install-hook:
 	libname=libsystemd-login.so && $(move-to-rootlibdir)
@@ -4452,6 +4445,13 @@
 
 endif
 
+pkgconfiglib_DATA += \
+	src/compat-libs/libsystemd-journal.pc \
+	src/compat-libs/libsystemd-login.pc \
+	src/compat-libs/libsystemd-id128.pc \
+	src/compat-libs/libsystemd-daemon.pc
+
+
 EXTRA_DIST += \
 	src/compat-libs/libsystemd-journal.pc.in \
 	src/compat-libs/libsystemd-login.pc.in \
diff -ru systemd-209/Makefile.in systemd-209-modified/Makefile.in
--- systemd-209/Makefile.in	2014-02-20 02:09:23.640888214 +0100
+++ systemd-209-modified/Makefile.in	2014-02-20 19:23:11.890184699 +0100
@@ -1068,11 +1068,11 @@
 @ENABLE_COMPAT_LIBS_TRUE@	libsystemd-id128.la \
 @ENABLE_COMPAT_LIBS_TRUE@	libsystemd-daemon.la
 
-@ENABLE_COMPAT_LIBS_TRUE@am__append_236 = \
-@ENABLE_COMPAT_LIBS_TRUE@	src/compat-libs/libsystemd-journal.pc \
-@ENABLE_COMPAT_LIBS_TRUE@	src/compat-libs/libsystemd-login.pc \
-@ENABLE_COMPAT_LIBS_TRUE@	src/compat-libs/libsystemd-id128.pc \
-@ENABLE_COMPAT_LIBS_TRUE@	src/compat-libs/libsystemd-daemon.pc
+am__append_236 = \
+	src/compat-libs/libsystemd-journal.pc \
+	src/compat-libs/libsystemd-login.pc \
+	src/compat-libs/libsystemd-id128.pc \
+	src/compat-libs/libsystemd-daemon.pc
 
 @ENABLE_COMPAT_LIBS_TRUE@am__append_237 = compat-lib-install-hook
 @ENABLE_COMPAT_LIBS_TRUE@am__append_238 = compat-lib-uninstall-hook
