From 13259a28a6ace55f9397b2066173fecc4b296833 Mon Sep 17 00:00:00 2001
From: Michael Marineau <michael.marineau@coreos.com>
Date: Thu, 9 Jul 2015 13:13:33 -0700
Subject: [PATCH] Bail out on --save-temps, an alias for -save-temps

---
 NEWS.txt  | 2 ++
 compopt.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/NEWS.txt b/NEWS.txt
index c1e121d..19b0ae2 100644
--- a/NEWS.txt
+++ b/NEWS.txt
@@ -18,6 +18,8 @@ Bug fixes
 
 - Avoid calling `exit()` inside an exit handler.
 
+- Bail out on compiler option --save-temps in addition to -save-temps.
+
 
 ccache 3.2.2
 ------------
diff --git a/compopt.c b/compopt.c
index 26e1591..0fc9161 100644
--- a/compopt.c
+++ b/compopt.c
@@ -33,6 +33,7 @@ struct compopt {
 
 static const struct compopt compopts[] = {
 	{"--param",         TAKES_ARG},
+	{"--save-temps",    TOO_HARD},
 	{"--serialize-diagnostics", TAKES_ARG | TAKES_PATH},
 	{"-A",              TAKES_ARG},
 	{"-D",              AFFECTS_CPP | TAKES_ARG | TAKES_CONCAT_ARG},
-- 
2.3.6

