--- fontconfig-2.7.1-orig/conf.d/30-metric-aliases.conf	2009-06-28 10:48:47.000000000 -0700
+++ fontconfig-2.7.1/conf.d/30-metric-aliases.conf	2010-08-02 13:34:53.000000000 -0700
@@ -29,6 +29,15 @@
 			Albany AMT
 			Thorndale AMT
 			Cumberland AMT
+                Chrom*OS fonts:
+			Arimo
+			Tinos
+			Cousine
+		Ascender fonts (eval):
+			Ascender Sans
+			Ascender Serif
+			Ascender Sans Mono
+			Ascender Symbol
 
 	     Of these, URW fonts are design compatible with PostScrict fonts,
 	     and the Liberation, StarOffice, and AMT ones are compatible with
@@ -75,6 +84,8 @@
 
 	<!-- Microsoft -->
 	<alias binding="same">
+	  <family>Arimo</family>
+	  <family>Ascender Sans</family>
 	  <family>Liberation Sans</family>
 	  <family>Albany</family>
 	  <family>Albany AMT</family>
@@ -84,6 +95,8 @@
 	</alias>
 
 	<alias binding="same">
+	  <family>Tinos</family>
+	  <family>Ascender Serif</family>
 	  <family>Liberation Serif</family>
 	  <family>Thorndale</family>
 	  <family>Thorndale AMT</family>
@@ -93,6 +106,8 @@
 	</alias>
 
 	<alias binding="same">
+	  <family>Cousine</family>
+	  <family>Ascender Sans Mono</family>
 	  <family>Liberation Mono</family>
 	  <family>Cumberland</family>
 	  <family>Cumberland AMT</family>
@@ -101,6 +116,13 @@
 	  </default>
 	</alias>
 
+	<alias binding="same">
+	  <family>Ascender Symbol</family>
+	  <default>
+	  <family>Symbol</family>
+	  </default>
+	</alias>
+
 
 
 <!-- Accept the other group as fallback -->
@@ -183,6 +205,8 @@
 	<alias binding="same">
 	  <family>Arial</family>
 	  <accept>
+	  <family>Arimo</family>
+	  <family>Ascender Sans</family>
 	  <family>Liberation Sans</family>
 	  <family>Albany</family>
 	  <family>Albany AMT</family>
@@ -192,6 +216,8 @@
 	<alias binding="same">
 	  <family>Times New Roman</family>
 	  <accept>
+	  <family>Tinos</family>
+	  <family>Ascender Serif</family>
 	  <family>Liberation Serif</family>
 	  <family>Thorndale</family>
 	  <family>Thorndale AMT</family>
@@ -201,6 +227,8 @@
 	<alias binding="same">
 	  <family>Courier New</family>
 	  <accept>
+	  <family>Cousine</family>
+	  <family>Ascender Sans Mono</family>
 	  <family>Liberation Mono</family>
 	  <family>Cumberland</family>
 	  <family>Cumberland AMT</family>
