diff -ru git-1.6.6.1.orig/compat/inet_ntop.c git-1.6.6.1/compat/inet_ntop.c
--- git-1.6.6.1.orig/compat/inet_ntop.c	2010-03-09 04:39:03 -0800
+++ git-1.6.6.1/compat/inet_ntop.c	2010-03-09 04:40:50 -0800
@@ -169,6 +169,8 @@
 }
 #endif
 
+/* conflicts with interix' headers... */
+#ifndef __INTERIX
 /* char *
  * inet_ntop(af, src, dst, size)
  *	convert a network format address to presentation format.
@@ -197,3 +199,4 @@
 	}
 	/* NOTREACHED */
 }
+#endif
